---
title: "ANCOVA of protein gorups and pSites"
output: html_notebook
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

```{r setup, include=FALSE}
knitr::opts_chunk$set(collapse=TRUE, comment="##", fig.retina=2, fig.path = "README_figs/README-")
```

Load packages
```{r, include=FALSE}
library(dplyr)
library(ggplot2)
library(reshape2)
library(tidyverse)
library(GGally)
library(ggpubr)
library(ggrepel)
library(qvalue)
library(gridExtra)
library(devtools)
library(ggfortify)
library(limma)
library(plotly)
library(broom)
library(rmotifx)
library(pheatmap)
```

Read in MASTER protein groups
```{r}
#read in the master table: 
master <- read_csv("./tables/master_tidy_protein.csv")
```

T-test for differential expressing proteins for each time point
```{r}
#prepare tables for statistical analysis
master_matched <- master %>%
  #remove the sudocounts 1 that has been added at the beginning
  mutate(intensity = intensity - 1) %>%
  melt() %>%
  dplyr::filter(genotype %in% c("rim15KO","WT")) %>%
# remove data whose values are 0 across all replicates
  group_by(`Protein IDs`, `Gene names`, genotype, nutrient, samplingTime, variable) %>%
  mutate(total = sum(value)) %>%
  dplyr::filter(total > 0.000000e+00) %>%
  #only keep proteins who has valid measurements in both nutritional conditions
  group_by(`Protein IDs`,`Gene names`, samplingTime, replicate, genotype, variable) %>%
  mutate(nutrient_number = n()) %>%
  dplyr::filter(nutrient_number == 2) %>%
  #only keep proteins who has valid measurements in both genotypes
  group_by(`Protein IDs`, `Gene names`, nutrient, samplingTime, replicate, variable) %>%
  mutate(genotype_number = n()) %>%
  dplyr::filter(genotype_number==2) %>%
  #only keep proteins who has valid measurements in more than 2 replicates
  group_by(`Protein IDs`, `Gene names`, genotype, nutrient, samplingTime, variable) %>%
  mutate(rep_numner = n()) %>%
  dplyr::filter(rep_numner >= 2) %>%
  #only keep proteins who has valid measurements in 4 timepoints
  group_by(`Protein IDs`, `Gene names`, nutrient, replicate, genotype, variable) %>%
  mutate(time_number = n()) %>%
  dplyr::filter(time_number == 4) %>%
  select(-c("nutrient_number","genotype_number","time_number","total")) %>%
  #remove the values whose has invalid read across all timepoints in each condition
  group_by(`Protein IDs`,`Gene names`, genotype, nutrient, replicate, variable) %>%
  #count the number of na values
  mutate(ind = sum(is.na(value))) %>% 
  #remove the proteins who has NA value in all conditions and samples
  dplyr::filter(!any(ind == 4)) %>%
  select(-ind)

t.test <- master_matched  %>%
  dplyr::filter(`Protein IDs` != "P38109") %>%
  group_by(`Protein IDs`, `Gene names`, nutrient, samplingTime, variable) %>%
  summarise_each(funs(t.test(log2(.[genotype == "WT"]), log2(.[genotype == "rim15KO"]), var.equal=TRUE)$p.value), pvalue = value) #%>%

t.test$qvalue <- p.adjust(t.test$pvalue,method="BH") 

#caculate log2FC 
log2FC <- master_matched  %>%
  dplyr::filter(`Protein IDs` != "P38109") %>%
    group_by(`Protein IDs`, `Gene names`, nutrient, samplingTime, variable) %>%
  summarise_each(funs(log2(mean(.[genotype=="WT"], na.rm=TRUE)/mean(.[genotype == "rim15KO"], na.rm=TRUE))), log2FC = value)

#combine log2FC and t.test results
DE_perTime <- left_join(t.test, log2FC) %>%
    mutate(qvalue = round(qvalue, digits = 4))

#compare the pair-wise correlation of DE on different data
DE_perTime %>% 
  select(-c("pvalue","qvalue")) %>%
  spread(key = variable, value = log2FC) %>%
  ggplot(aes(x=ratio, y = normalized_ratio)) +
  geom_point() +
  facet_grid(nutrient~samplingTime)

DE_perTime %>% 
  select(-c("pvalue","qvalue")) %>%
  spread(key = variable, value = log2FC) %>%
  ggplot(aes(x=ratio, y = intensity)) +
  geom_point() +
  facet_grid(nutrient~samplingTime)

DE_perTime %>% 
  select(-c("pvalue","qvalue")) %>%
  spread(key = variable, value = log2FC) %>%
  ggplot(aes(x=ratio, y = normalized_intensity)) +
  geom_point() +
  facet_grid(nutrient~samplingTime)

DE_perTime %>% 
  select(-c("pvalue","qvalue")) %>%
  spread(key = variable, value = log2FC) %>%
  ggplot(aes(x=intensity, y = normalized_intensity)) +
  geom_point() +
  facet_grid(nutrient~samplingTime)

DE_perTime %>% 
  select(-c("pvalue","qvalue")) %>%
  spread(key = variable, value = log2FC) %>%
  ggplot(aes(x=normalized_ratio, y = normalized_intensity)) +
  geom_point() +
  facet_grid(nutrient~samplingTime)
```

ANCOVA for deferentially expressed proteins across timepoint between genotypes 
```{r}
genotype_matched <- proteome %>%
  filter(genotype %in% c("rim15KO","WT")) %>%
  group_by(`Gene names`, samplingTime, replicate, genotype) %>%
  mutate(nutrient_number = n()) %>%
  filter(nutrient_number == 2) %>%
  group_by(`Gene names`, nutrient, samplingTime, replicate) %>%
  mutate(genotype_number = n()) %>%
  filter(genotype_number==2) %>%
  group_by(`Gene names`, nutrient, samplingTime, genotype) %>%
  mutate(rep_numner = n()) %>%
  filter(rep_numner == 3) %>%
  group_by(`Gene names`, nutrient, replicate, genotype) %>%
  mutate(time_number = n()) %>%
  filter(time_number == 4) %>%
  #remove data with all 0s
  

ancova_genotype <- genotype_matched %>% 
  mutate(samplingTime = substr(samplingTime, 2, nchar(samplingTime))) %>%
  mutate(samplingTime = as.numeric(samplingTime)) %>%
  dplyr::group_by(nutrient, `Gene names`) %>%
  do(tidy(lm(log10(normalized_intensity) ~ samplingTime * genotype, .))) %>% 
  dplyr::filter(!is.na(p.value)) #%>%

#add q-value
ancova_genotype$qvalue <- qvalue(ancova_genotype$p.value, fdr.level=0.05, pi0.method="bootstrap")$qvalues

#check the distribution of statistics
ancova_genotype %>% 
  ggplot(aes(x=p.value, col = nutrient, fill = nutrient, alpha = 0.2)) + 
  geom_histogram(aes(y=..density..),  position="identity", alpha=0.1)+
  facet_grid( ~ term) +
  scale_color_manual(values=c("#a6beff", "#ff7369"))+
  scale_fill_manual(values=c("#a6beff",  "#ff7369")) +
  theme_classic2() +
  theme(legend.position = "bottom") +
  ggtitle("pvalue")

#check the distribution of coefficient
ancova_genotype %>% 
  ggplot(aes(x=estimate, col = nutrient, fill = nutrient, alpha = 0.2)) + 
  geom_histogram(aes(y=..density..),  position="identity", alpha=0.1)+
  facet_grid( ~ term) +
  scale_color_manual(values=c("#a6beff", "#ff7369"))+
  scale_fill_manual(values=c("#a6beff",  "#ff7369")) +
  theme_classic2() +
  theme(legend.position = "bottom") +
  ggtitle("Coefficient")

#check the distribution of qvalue
ancova_genotype %>% 
  ggplot(aes(x=qvalue, col = nutrient, fill = nutrient, alpha = 0.2)) + 
  geom_histogram(aes(y=..density..),  position="identity", alpha=0.1)+
  facet_grid( ~ term) +
  scale_color_manual(values=c("#a6beff", "#ff7369"))+
  scale_fill_manual(values=c("#a6beff",  "#ff7369")) +
  theme_classic2() +
  theme(legend.position = "bottom")+
  ggtitle("p.adj (FDR = 0.05)")

#look at the significantly differently expressed genes
sig <- ancova_genotype %>% filter(term == "samplingTime:genotypeWT" & qvalue < 0.05)
```

ANCOVA for deferentially expressed proteins across timepoint between conditions 
```{r}
ancova_nutrient <- genotype_matched %>% 
  mutate(samplingTime = substr(samplingTime, 2, nchar(samplingTime))) %>%
  mutate(samplingTime = as.numeric(samplingTime)) %>%
  dplyr::group_by(genotype, `Gene names`) %>%
  do(tidy(lm(log10(normalized_intensity) ~ samplingTime * nutrient, .))) %>% 
  dplyr::filter(!is.na(p.value)) #%>%

#add q-value
ancova_nutrient$qvalue <- qvalue(ancova_nutrient$p.value, fdr.level=0.05, pi0.method="bootstrap")$qvalues

#check the distribution of statistics
ancova_nutrient %>% 
  ggplot(aes(x=p.value, col = genotype, fill = genotype, alpha = 0.2)) + 
  geom_histogram(aes(y=..density..),  position="identity", alpha=0.1)+
  facet_grid( ~ term) +
  scale_color_manual(values=c("#a6beff", "#ff7369"))+
  scale_fill_manual(values=c("#a6beff",  "#ff7369")) +
  theme_classic2() +
  theme(legend.position = "bottom") +
  ggtitle("pvalue")

#check the distribution of coefficient
ancova_nutrient %>% 
  ggplot(aes(x=estimate, col = genotype, fill = genotype, alpha = 0.2)) + 
  geom_histogram(aes(y=..density..),  position="identity", alpha=0.1)+
  facet_grid( ~ term) +
  scale_color_manual(values=c("#a6beff", "#ff7369"))+
  scale_fill_manual(values=c("#a6beff",  "#ff7369")) +
  theme_classic2() +
  theme(legend.position = "bottom") +
  ggtitle("Coefficient")

#check the distribution of qvalue
ancova_nutrient %>% 
  ggplot(aes(x=qvalue, col = genotype, fill = genotype, alpha = 0.2)) + 
  geom_histogram(aes(y=..density..),  position="identity", alpha=0.1)+
  facet_grid(data ~ term) +
  scale_color_manual(values=c("#a6beff", "#ff7369"))+
  scale_fill_manual(values=c("#a6beff",  "#ff7369")) +
  theme_classic2() +
  theme(legend.position = "bottom")+
  ggtitle("p.adj (FDR = 0.05)")

#look at the significantly differently expressed genes
sig_nutrient <- ancova_nutrient %>% filter(term == "samplingTime:nutrientP" & qvalue < 0.05)
dim(sig_nutrient)
head(sig_nutrient)
```

Three-way ANCOVA for differentially expressed proteins across timepoints, bt conditions, and bt genotypes
```{r}
#three way anova using samplingTime, genotype and replicate as variables 
ancova_geno_nutrient <- genotype_matched %>% 
  mutate(samplingTime = substr(samplingTime, 2, nchar(samplingTime))) %>%
  mutate(samplingTime = as.numeric(samplingTime)) %>%
  dplyr::group_by(`Gene names`) %>%
  do(tidy(aov(log10(normalized_intensity) ~ samplingTime * genotype * nutrient, .))) %>% 
  dplyr::filter(!is.na(p.value))
#add q-value
ancova_geno_nutrient$qvalue <- qvalue(ancova_geno_nutrient$p.value, fdr.level=0.05, pi0.method="bootstrap")$qvalues 

#check the distribution of statistics
ancova_geno_nutrient %>% 
  ggplot(aes(x=p.value, col = "#ff7369", alpha = 0.2)) + 
  geom_histogram(aes(y=..density..), col = "#ff7369", position="identity", alpha=0.1)+
  facet_grid( ~ term) +
   theme_classic2() +
  theme(legend.position = "bottom") +
  ggtitle("pvalue")

#check the distribution of coefficient
ancova_geno_nutrient %>% 
  ggplot(aes(x=sumsq, col = "#ff7369", alpha = 0.2)) + 
  geom_histogram(aes(y=..density..), col = "#ff7369", position="identity", alpha=0.1)+
  facet_grid( ~ term) +
  theme_classic2() +
  theme(legend.position = "bottom") +
  ggtitle("sum of square root")

#check the distribution of qvalue
ancova_geno_nutrient %>% 
  ggplot(aes(x=qvalue, col = "#ff7369", alpha = 0.2)) + 
  geom_histogram(aes(y=..density..), col = "#ff7369", position="identity", alpha=0.1)+
  facet_grid( ~ term) +
  theme_classic2() +
  theme(legend.position = "bottom")+
  ggtitle("p.adj (FDR = 0.05)")

sig_geno <- ancova_geno_nutrient %>% filter(term == "samplingTime:genotype" & qvalue < 0.05)
sig_nutri <- ancova_geno_nutrient %>% filter(term == "samplingTime:nutrient" & qvalue < 0.05)
```



