---
title: "PCA and DE analysis for proteins"
output: 
  html_document: 
    keep_md: true
---

## Chrunks

* [Load libraries]
* [Read in tables]
* [Proteomic data preparation]
  * [QC.1 - Intensity and normalized intensity]
    * [Data preparation]
    * [Box plot]
    * [Scatter plot]
  * [QC.2 - Ratio and normalized ratio]
    * [Data preparation]
    * [Box plot]
    * [Scatter plot]
  * [Combine ratios and intensity into one table]
* [PCA analysis for each data type]
* [T-test for DE at each timepoint]


## Load libraries
```{r, message=FALSE}
library(dplyr)
library(qvalue)
library(ggplot2)
library(reshape2)
library(tidyverse)
library(GGally)
library(ggpubr)
library(ggrepel)
library(gridExtra)
library(devtools)
library(ggfortify)
library(limma)
library(plotly)
library(broom)
library(pheatmap)
library(psych)  # geometric mean calculation
library(RColorBrewer)
```


## Read in tables 
```{r, echo = FALSE, message=FALSE}
proteinGroups <- read_csv("./data/proteinGroups.csv")
head(proteinGroups)
dim(proteinGroups)
```


## Proteomic data preparation
* This code extract the information from proteinGroup file generated in MaxQuant and put them in a tidy format
  * intensity
    * Run Anders & Huber, 2010 method to get normalized intensity for each protein
  * ratio
  * normalized_ratio

### QC.1 - Intensity and normalized intensity
#### Data preparation
```{r, echo = FALSE, warning = FALSE, message=FALSE}
#extract the protome info
intensity <- proteinGroups %>%
  dplyr::select(`Protein IDs`, `Gene names`,`Protein names`, `Fasta headers`, contains("proteome"),) %>%
  dplyr::select(`Protein IDs`, `Gene names`,`Protein names`, `Fasta headers`, contains("Intensity")) #%>%
  #remove nitrogen sample for downstream analysis
  #dplyr::select(-matches("_N_"))

#extract data for time 0
#nitrogen
T00_N <- intensity %>% dplyr::select(`Protein IDs`, `Gene names`,`Protein names`, `Fasta headers`, contains("T00")) %>% 
  rename(`Intensity proteome_N_T00_R1` = `Intensity proteome_C_T00_R1`,
         `Intensity proteome_N_T00_R2` = `Intensity proteome_C_T00_R2`,
         `Intensity proteome_N_T00_R3` = `Intensity proteome_C_T00_R3`,
         `Intensity L proteome_N_T00_R1` = `Intensity L proteome_C_T00_R1`,
         `Intensity L proteome_N_T00_R2` = `Intensity L proteome_C_T00_R2`,
         `Intensity L proteome_N_T00_R3` = `Intensity L proteome_C_T00_R3`,
         `Intensity M proteome_N_T00_R1` = `Intensity M proteome_C_T00_R1`,
         `Intensity M proteome_N_T00_R2` = `Intensity M proteome_C_T00_R2`,
         `Intensity M proteome_N_T00_R3` = `Intensity M proteome_C_T00_R3`,
         `Intensity H proteome_N_T00_R1` = `Intensity H proteome_C_T00_R1`,
         `Intensity H proteome_N_T00_R2` = `Intensity H proteome_C_T00_R2`,
         `Intensity H proteome_N_T00_R3` = `Intensity H proteome_C_T00_R3`)
#phosphorous
T00_P <- intensity %>% dplyr::select(`Protein IDs`, `Gene names`,`Protein names`, `Fasta headers`, contains("T00")) %>% 
  rename(`Intensity proteome_P_T00_R1` = `Intensity proteome_C_T00_R1`,
         `Intensity proteome_P_T00_R2` = `Intensity proteome_C_T00_R2`,
         `Intensity proteome_P_T00_R3` = `Intensity proteome_C_T00_R3`,
         `Intensity L proteome_P_T00_R1` = `Intensity L proteome_C_T00_R1`,
         `Intensity L proteome_P_T00_R2` = `Intensity L proteome_C_T00_R2`,
         `Intensity L proteome_P_T00_R3` = `Intensity L proteome_C_T00_R3`,
         `Intensity M proteome_P_T00_R1` = `Intensity M proteome_C_T00_R1`,
         `Intensity M proteome_P_T00_R2` = `Intensity M proteome_C_T00_R2`,
         `Intensity M proteome_P_T00_R3` = `Intensity M proteome_C_T00_R3`,
         `Intensity H proteome_P_T00_R1` = `Intensity H proteome_C_T00_R1`,
         `Intensity H proteome_P_T00_R2` = `Intensity H proteome_C_T00_R2`,
         `Intensity H proteome_P_T00_R3` = `Intensity H proteome_C_T00_R3`)

intensity<- inner_join(inner_join(intensity, T00_N), T00_P)

intensity_tidy <- intensity %>% 
  dplyr::select(`Protein IDs`, `Gene names`,`Protein names`, `Fasta headers`, matches(" H | M | L ")) %>%
  melt() %>%
  separate(variable, c("signal","channel","sampleNum"), sep = " ") %>%
  separate(sampleNum, c("omic","nutrient","samplingTime","replicate"), sep = "_") %>%
  dplyr::select(-c("signal","omic")) %>%
  #rename value by intensity
  rename (intensity = value)

#reverse the channel information for replicate 2
rep2 <- intensity_tidy %>%
  dplyr::filter(replicate == "R2") %>% 
  mutate(genotype = case_when(
     channel == "H"  ~ "rim15KO", 
     channel == "M"  ~ "WT",
     channel == "L"  ~ "spike-in"))

#add information for genotype based on channels
intensity_rest <- intensity_tidy %>% 
  dplyr::filter(replicate != "R2") %>%
   mutate(genotype = case_when(
     channel == "M"  ~ "rim15KO", 
     channel == "H"  ~ "WT",
     channel == "L"  ~ "spike-in"))

#combined new tables into one
intensity_tidy_re <- bind_rows(intensity_rest, rep2)

##remove nitrogen samples for size factor estimation 
# note that size factor was only estimated based on spike-ins
intensity_geoMean <- intensity_tidy_re %>%
  dplyr::filter(genotype == "spike-in") %>%
  group_by(`Protein IDs`, `Gene names`, `Protein names`, `Fasta headers`) %>%
  summarize(ref = geometric.mean(intensity)) #%>%
  #drop_na()

#filter out proteins whose genometric mean across all samples are smaller than 10000
intensity_geoMean <- intensity_geoMean %>% filter(ref > 10000)

intensity_geoMean_ratio2ref <- right_join(intensity_tidy_re, intensity_geoMean) %>%
  #divide each intensity/protein group by its geometric mean
  mutate(ratio2ref = intensity / ref)

size_factor <- intensity_geoMean_ratio2ref %>%
  #estimate the size factor for each condition
  group_by(nutrient, samplingTime, replicate, genotype) %>%
  summarize(sizeFactor = median (ratio2ref , na.rm = TRUE))

intensity_geoMean_normed <- left_join(intensity_geoMean_ratio2ref, size_factor, by = c("nutrient","samplingTime","replicate","genotype")) %>%
  mutate (normed_intensity = intensity/sizeFactor) %>%
  select (-c("ref","ratio2ref","sizeFactor")) #%>%
#  filter(genotype != "spike-in")

#visualization
##order genotypes and channels in plot
intensity_geoMean_normed$genotype <- factor(intensity_geoMean_normed$genotype, levels = c("WT","rim15KO","spike-in"))
intensity_geoMean_normed$channel <- factor(intensity_geoMean_normed$channel, levels = c("H","M","L"))
```

#### Box plot
```{r, echo = FALSE, warning = FALSE, message=FALSE}
ggplot(intensity_geoMean_normed %>% filter(intensity > 0), aes(x=samplingTime, y=intensity, fill = channel)) + 
  geom_point(aes(colour = channel, x = samplingTime), position=position_jitterdodge(dodge.width = 0.9), size = 0.3, alpha = 0.3) +
  geom_boxplot(position=position_dodge(width = 0.9), outlier.shape = NA)+
  #
  facet_grid(replicate~nutrient) +
  theme_classic() +
  scale_color_manual(values=c("#fcb86d","#7fd3cf","#babcbf"))+
  scale_fill_manual(values=c("#fcb86d","#7fd3cf","#babcbf"))+
  scale_y_log10() +
  ylab("Intensity")

ggplot(intensity_geoMean_normed %>% filter(intensity > 0), aes(x=samplingTime, y=intensity, fill = genotype)) + 
  #scale_x_discrete() + 
  #geom_jitter(aes(colour = genotype, x = samplingTime), 
   #  position = position_jitter(width = .9), alpha = 0.5) +
  geom_point(aes(colour = genotype, x = samplingTime), position=position_jitterdodge(dodge.width = 0.9), size = 0.3, alpha = 0.1) +
  geom_boxplot(position=position_dodge(width = 0.9), outlier.shape = NA)+
  #
  facet_grid(replicate~nutrient) +
  theme_classic() +
  scale_color_manual(values=c("#13b1f0","#e7c100","#eb5534"))+
  scale_fill_manual(values=c("#13b1f0","#e7c100","#eb5534"))+
  scale_y_log10() +
  ylab("Intensity")

ggplot(intensity_geoMean_normed %>% filter(normed_intensity > 0), aes(x=samplingTime, y=normed_intensity, fill = channel)) + 
  geom_point(aes(colour = channel, x = samplingTime), position=position_jitterdodge(dodge.width = 0.9), size = 0.3, alpha = 0.3) +
  geom_boxplot(position=position_dodge(width = 0.9), outlier.shape = NA)+
  #
  facet_grid(replicate~nutrient) +
  theme_classic() +
  scale_color_manual(values=c("#fcb86d","#7fd3cf","#babcbf"))+
  scale_fill_manual(values=c("#fcb86d","#7fd3cf","#babcbf"))+
  scale_y_log10() +
  ylab("Normalized intensity")

ggplot(intensity_geoMean_normed %>% filter(normed_intensity > 0), aes(x=samplingTime, y=normed_intensity, fill = genotype)) + 
    geom_point(aes(colour = genotype, x = samplingTime), position=position_jitterdodge(dodge.width = 0.9), size = 0.3, alpha = 0.1) +
  geom_boxplot(position=position_dodge(0.9), outlier.shape = NA)+#, outlier.shape = NA) +
  #
  facet_grid(replicate~nutrient) +
  theme_classic() +
  scale_color_manual(values=c("#13b1f0","#e7c100","#eb5534")) +
  scale_fill_manual(values=c("#13b1f0","#e7c100","#eb5534")) +
  scale_y_log10() +
  ylab("Normalized intensity")
```

#### Scatter plot
```{r, echo = FALSE, warning = FALSE, message=FALSE}
normed_spread <- intensity_geoMean_normed %>% 
  select(-c("intensity","genotype")) %>%
  filter(normed_intensity > 0) %>%
  spread(key = replicate, value = normed_intensity) 
  
ggscatter(normed_spread, x = "R1", y = "R2", alpha = 0.3, merge = TRUE, size=0.8, col= "channel", shape = "channel", palette = c("#fcb86d","#7fd3cf","#babcbf")) + 
  stat_cor(aes(color = channel, group = channel), method = "pearson", label.x.npc = "left", label.y.npc = "top", size=2) +
  facet_grid(nutrient~samplingTime, scale='free') +
  theme_classic()+
  scale_y_log10()+
  scale_x_log10()+
  xlab("Rep1")+
  ylab("Rep2")+
  ggtitle("Scatter plot for Rep1 vs Rep2 (Normalized intensity)")

ggscatter(normed_spread, x = "R1", y = "R3", alpha = 0.3, merge = TRUE, size=0.8, col= "channel", shape = "channel", palette = c("#fcb86d","#7fd3cf","#babcbf")) + 
  stat_cor(aes(color = channel, group = channel), method = "pearson", label.x.npc = "left", label.y.npc = "top", size=2) +
  facet_grid(nutrient~samplingTime, scale='free') +
  theme_classic()+
  scale_y_log10()+
  scale_x_log10()+
  xlab("Rep1")+
  ylab("Rep3")+
  ggtitle("Scatter plot for Rep1 vs Rep2 (Normalized intensity)")

ggscatter(normed_spread, x = "R3", y = "R2", alpha = 0.3, merge = TRUE, size=0.8, col= "channel", shape = "channel", palette = c("#fcb86d","#7fd3cf","#babcbf")) + 
  stat_cor(aes(color = channel, group = channel), method = "pearson", label.x.npc = "left", label.y.npc = "top", size=2) +
  facet_grid(nutrient~samplingTime, scale='free') +
  theme_classic()+
  scale_y_log10()+
  scale_x_log10()+
  xlab("Rep3")+
  ylab("Rep2")+
  ggtitle("Scatter plot for Rep1 vs Rep2 (Normalized intensity)")
```

### QC.2 - Ratio and normalized ratio
#### Data preparation
```{r, echo = FALSE, warning = FALSE, message=FALSE}
##deal with ratios and then combine it with the normalized intensity data
#extract the ratio info
ratio <- proteinGroups %>% 
  #select columns that contains the useful information - based on the experiments setups
  dplyr::select(`Protein IDs`, `Gene names`,`Protein names`, `Fasta headers`, contains("Ratio")) %>%
  dplyr::select(`Protein IDs`, `Gene names`, `Protein names`, `Fasta headers`, contains("proteome")) %>%
  #remove columns that is not important
  #dplyr::select(-matches("_N_")) %>% 
  dplyr::select(-contains("variability")) %>%
  dplyr::select(-contains("normalized")) %>%
  dplyr::select(-contains("count")) %>%
  dplyr::select(-contains("iso-count")) %>%
  dplyr::select(-contains("type"))

#extract data for time 0
##nitrogen
T00_N <- ratio %>% dplyr::select(`Protein IDs`, `Gene names`,`Protein names`, `Fasta headers`, contains("T00")) %>% 
  dplyr::rename(`Ratio M/L proteome_N_T00_R1` = `Ratio M/L proteome_C_T00_R1`,
         `Ratio M/L proteome_N_T00_R2` = `Ratio M/L proteome_C_T00_R2`,
         `Ratio M/L proteome_N_T00_R3` = `Ratio M/L proteome_C_T00_R3`,
         `Ratio H/L proteome_N_T00_R1` = `Ratio H/L proteome_C_T00_R1`,
         `Ratio H/L proteome_N_T00_R2` = `Ratio H/L proteome_C_T00_R2`,
         `Ratio H/L proteome_N_T00_R3` = `Ratio H/L proteome_C_T00_R3`,
         `Ratio H/M proteome_N_T00_R1` = `Ratio H/M proteome_C_T00_R1`,
         `Ratio H/M proteome_N_T00_R2` = `Ratio H/M proteome_C_T00_R2`,
         `Ratio H/M proteome_N_T00_R3` = `Ratio H/M proteome_C_T00_R3`)

##phosphorous
T00_P <- ratio %>% dplyr::select(`Protein IDs`, `Gene names`,`Protein names`, `Fasta headers`, contains("T00")) %>% 
  dplyr::rename(`Ratio M/L proteome_P_T00_R1` = `Ratio M/L proteome_C_T00_R1`,
         `Ratio M/L proteome_P_T00_R2` = `Ratio M/L proteome_C_T00_R2`,
         `Ratio M/L proteome_P_T00_R3` = `Ratio M/L proteome_C_T00_R3`,
         `Ratio H/L proteome_P_T00_R1` = `Ratio H/L proteome_C_T00_R1`,
         `Ratio H/L proteome_P_T00_R2` = `Ratio H/L proteome_C_T00_R2`,
         `Ratio H/L proteome_P_T00_R3` = `Ratio H/L proteome_C_T00_R3`,
         `Ratio H/M proteome_P_T00_R1` = `Ratio H/M proteome_C_T00_R1`,
         `Ratio H/M proteome_P_T00_R2` = `Ratio H/M proteome_C_T00_R2`,
         `Ratio H/M proteome_P_T00_R3` = `Ratio H/M proteome_C_T00_R3`)

ratio_add<- dplyr::inner_join(dplyr::inner_join(ratio, T00_P), T00_N)
  
#reshape the table into a tidy format 
ratio_tidy <- melt(ratio_add) %>%
  separate(variable, c("signal","channel","sampleNum"), sep = " ") %>%
  separate(sampleNum, c("omic","nutrient","samplingTime","replicate"), sep = "_") %>%
  dplyr::select(-c("signal","omic"))  %>%
  rename(ratio = value) %>%
  #select only wild type and ko ratios individually
  dplyr::filter(channel %in% c("H/L","M/L"))

##reversed build-in ratio by genotypes
##switch H/L to M/L
rep2 <- ratio_tidy %>% 
  filter(replicate == "R2" & nutrient == "P") %>%
  mutate(genotype = case_when(
    channel == "H/L"  ~ "rim15KO", 
    channel == "M/L"  ~ "WT")) %>%
  mutate(channel=replace(channel, channel=="M/L", "M")) %>%
  mutate(channel=replace(channel, channel=="H/L", "H"))

#join the revised replicate 2 with the other data from replicate 1&3
#remove old data
new_re <- anti_join(ratio_tidy , ratio_tidy %>% dplyr::filter (nutrient == "P" & replicate == "R2"))

ratio_rest <- new_re %>%
  mutate(genotype = case_when(
    channel == "M/L"  ~ "rim15KO", 
    channel == "H/L"  ~ "WT")) %>%
  mutate(channel=replace(channel, channel=="M/L", "M")) %>%
  mutate(channel=replace(channel, channel=="H/L", "H"))

ratio_tidy_re <- bind_rows(ratio_rest, rep2)

#extract the proteome info
ratio_normed <- proteinGroups %>% 
  #select columns that contains the useful information - based on the experiments setups
  dplyr::select(`Protein IDs`, `Gene names`,`Protein names`, `Fasta headers`, contains("Ratio")) %>%
  dplyr::select(`Protein IDs`, `Gene names`, `Protein names`, `Fasta headers`, contains("proteome")) %>%
  dplyr::select(`Protein IDs`, `Gene names`, `Protein names`, `Fasta headers`, contains("normalized")) %>%
  dplyr::select(-contains("variability")) %>%
  dplyr::select(-contains("count")) %>%
  dplyr::select(-contains("iso-count")) %>%
  dplyr::select(-contains("type"))

#extract data for time 0
T00_N <- ratio_normed %>% dplyr::select(`Protein IDs`, `Gene names`,`Protein names`, `Fasta headers`, contains("T00")) %>% 
  dplyr::rename(`Ratio M/L normalized proteome_N_T00_R1` = `Ratio M/L normalized proteome_C_T00_R1`,
         `Ratio M/L normalized proteome_N_T00_R2` = `Ratio M/L normalized proteome_C_T00_R2`,
         `Ratio M/L normalized proteome_N_T00_R3` = `Ratio M/L normalized proteome_C_T00_R3`,
         `Ratio H/L normalized proteome_N_T00_R1` = `Ratio H/L normalized proteome_C_T00_R1`,
         `Ratio H/L normalized proteome_N_T00_R2` = `Ratio H/L normalized proteome_C_T00_R2`,
         `Ratio H/L normalized proteome_N_T00_R3` = `Ratio H/L normalized proteome_C_T00_R3`,
         `Ratio H/M normalized proteome_N_T00_R1` = `Ratio H/M normalized proteome_C_T00_R1`,
         `Ratio H/M normalized proteome_N_T00_R2` = `Ratio H/M normalized proteome_C_T00_R2`,
         `Ratio H/M normalized proteome_N_T00_R3` = `Ratio H/M normalized proteome_C_T00_R3`)

T00_P <- ratio_normed %>% dplyr::select(`Protein IDs`, `Gene names`,`Protein names`, `Fasta headers`, contains("T00")) %>% 
  dplyr::rename(`Ratio M/L normalized proteome_P_T00_R1` = `Ratio M/L normalized proteome_C_T00_R1`,
         `Ratio M/L normalized proteome_P_T00_R2` = `Ratio M/L normalized proteome_C_T00_R2`,
         `Ratio M/L normalized proteome_P_T00_R3` = `Ratio M/L normalized proteome_C_T00_R3`,
         `Ratio H/L normalized proteome_P_T00_R1` = `Ratio H/L normalized proteome_C_T00_R1`,
         `Ratio H/L normalized proteome_P_T00_R2` = `Ratio H/L normalized proteome_C_T00_R2`,
         `Ratio H/L normalized proteome_P_T00_R3` = `Ratio H/L normalized proteome_C_T00_R3`,
         `Ratio H/M normalized proteome_P_T00_R1` = `Ratio H/M normalized proteome_C_T00_R1`,
         `Ratio H/M normalized proteome_P_T00_R2` = `Ratio H/M normalized proteome_C_T00_R2`,
         `Ratio H/M normalized proteome_P_T00_R3` = `Ratio H/M normalized proteome_C_T00_R3`)

ratio_normed_add<- dplyr::inner_join(dplyr::inner_join(ratio_normed, T00_P), T00_N)

#reshape the table into a tidy format 
ratio_normed_tidy <- melt(ratio_normed_add) %>%
  separate(variable, c("signal","channel","normed","sampleNum"), sep = " ") %>%
  separate(sampleNum, c("omic","nutrient","samplingTime","replicate"), sep = "_") %>%
  dplyr::select(-c("signal","omic", "normed"))  %>%
  rename(normed_ratio = value) %>%
  #select only wild type and ko ratios individually
  dplyr::filter(channel %in% c("H/L","M/L"))

##reversed build-in ratio by genotypes
##switch H/L to M/L
rep2 <- ratio_normed_tidy %>% 
  dplyr::filter(replicate == "R2" & nutrient == "P") %>%
  mutate(genotype = case_when(channel == "H/L"  ~ "rim15KO", channel == "M/L"  ~ "WT"))  %>%
  mutate(channel=replace(channel, channel=="M/L", "M")) %>%
  mutate(channel=replace(channel, channel=="H/L", "H"))

#join the revised replicate 2 with the other data from replicate 1&3
#remove old data
new_normed_re <- anti_join(ratio_normed_tidy,ratio_normed_tidy %>% dplyr::filter (nutrient == "P" & replicate == "R2"))

ratio_normed_rest <- new_normed_re %>%
  mutate(genotype = case_when(
     channel == "M/L"  ~ "rim15KO", 
     channel == "H/L"  ~ "WT"))  %>%
  mutate(channel=replace(channel, channel=="M/L", "M")) %>%
  mutate(channel=replace(channel, channel=="H/L", "H"))

ratio_normed_tidy_re <- bind_rows(ratio_normed_rest, rep2)

#reshape the core dataset for perseus 
  perseus <- ratio_normed_tidy_re %>% 
    select(`Protein IDs`,`Protein names`,`Gene names`,`Fasta headers`,
           genotype,nutrient,samplingTime,replicate,normed_ratio) %>%
    unite(cond, genotype, nutrient, samplingTime, replicate) %>%
    spread(key = cond, value = normed_ratio)

#replace NA value in Genenames column with protein names
  perseus$`Gene names` <- ifelse(is.na(perseus$`Gene names`), perseus$`Protein names`, perseus$`Gene names`) 
  write_csv(perseus,"./tables/normed_ratio_perseus.csv")

ratio_all<- inner_join(ratio_normed_tidy_re, ratio_tidy_re)

#visualization
ratio_all$genotype <- factor(ratio_all$genotype, levels = c("WT","rim15KO"))
ratio_all$channel <- factor(ratio_all$channel, levels = c("H","M"))

#save table
write_csv(ratio_all, "./tables/combined_ratio_proteom.csv")
```

#### Box plot
```{r, echo = FALSE, warning = FALSE, message=FALSE}
ggplot(ratio_all, aes(x=samplingTime, y=ratio, fill = channel)) + 
  geom_point(aes(colour = channel, x = samplingTime), position=position_jitterdodge(dodge.width = 0.9), alpha = 0.3) +
  geom_boxplot(position=position_dodge(width = 0.9), outlier.shape = NA,size = 0.4)+
  facet_grid(replicate~nutrient) +
  theme_classic() +
  scale_color_manual(values=c("#fcb86d","#7fd3cf"))+
  scale_fill_manual(values=c("#fcb86d","#7fd3cf"))+
  scale_y_log10() +
  theme_bw(base_size =10) +
  theme(panel.grid.minor = element_blank()) +
  xlab("Hours post nutrient removal") +
  ylab("Ratio relative to light channel")

ggplot(ratio_all, aes(x=samplingTime, y=ratio, fill = genotype)) + 
  geom_point(aes(colour = genotype, x = samplingTime), position=position_jitterdodge(dodge.width = 0.9), alpha = 0.3) +
  geom_boxplot(position=position_dodge(0.8), outlier.shape = NA) + #) +
  #
  facet_grid(replicate~nutrient) +
  theme_classic() +
  scale_y_log10() +
  scale_color_manual(values=c("#13b1f0","#e7c100","#eb5534"))+
  scale_fill_manual(values=c("#13b1f0","#e7c100","#eb5534"))+
  ylab("Ratio relative to L")

ggplot(ratio_all, aes(x=samplingTime, y=normed_ratio, fill = channel)) + 
  geom_point(aes(colour = channel, x = samplingTime), position=position_jitterdodge(dodge.width = 0.9), alpha = 0.3) +
  geom_boxplot(position=position_dodge(width = 0.9), outlier.shape = NA)+
  #
  facet_grid(replicate~nutrient) +
  theme_classic() +
  scale_color_manual(values=c("#f5a742","#57c275"))+
  scale_fill_manual(values=c("#f5a742","#57c275"))+
  scale_y_log10() +
  ylab("Normalized ratio relative to L")

ggplot(ratio_all, aes(x=samplingTime, y=normed_ratio, fill = genotype)) + 
  geom_point(aes(colour = genotype, x = samplingTime), position=position_jitterdodge(dodge.width = 0.9), alpha = 0.3) +
  geom_boxplot(position=position_dodge(0.8), outlier.shape = NA)+#, outlier.shape = NA) +
  #
  facet_grid(replicate~nutrient) +
  theme_classic() +
  scale_y_log10() +
  scale_color_manual(values=c("#13b1f0","#e7c100","#eb5534"))+
  scale_fill_manual(values=c("#13b1f0","#e7c100","#eb5534"))+
  ylab("Normalized ratio relative to L")
```

#### Scatter plot
```{r, echo = FALSE, warning = FALSE, message=FALSE}
normed_spread <- ratio_all %>% 
  select(-c("ratio","genotype")) %>%
  filter(normed_ratio > 0) %>%
  spread(key = replicate, value = normed_ratio) 
  
ggscatter(normed_spread, x = "R1", y = "R2", alpha = 0.3, merge = TRUE, size=0.8, col= "channel", shape = "channel", palette = c("#fcb86d","#7fd3cf","#babcbf")) + 
  stat_cor(aes(color = channel, group = channel), method = "pearson", label.x.npc = "left", label.y.npc = "top", size=2) +
  facet_grid(nutrient~samplingTime, scale='free') +
  theme_classic()+
  scale_y_log10()+
  scale_x_log10()+
  xlab("Rep1")+
  ylab("Rep2")+
  ggtitle("Scatter plot for Rep1 vs Rep2 (Normalized ratio)")

ggscatter(normed_spread, x = "R1", y = "R3", alpha = 0.3, merge = TRUE, size=0.8, col= "channel", shape = "channel", palette = c("#fcb86d","#7fd3cf","#babcbf")) + 
  stat_cor(aes(color = channel, group = channel), method = "pearson", label.x.npc = "left", label.y.npc = "top", size=2) +
  facet_grid(nutrient~samplingTime, scale='free') +
  theme_classic()+
  scale_y_log10()+
  scale_x_log10()+
  xlab("Rep1")+
  ylab("Rep3")+
  ggtitle("Scatter plot for Rep1 vs Rep2 (Normalized ratio)")

ggscatter(normed_spread, x = "R3", y = "R2", alpha = 0.3, merge = TRUE, size=0.8, col= "channel", shape = "channel", palette = c("#fcb86d","#7fd3cf","#babcbf")) + 
  stat_cor(aes(color = channel, group = channel), method = "pearson", label.x.npc = "left", label.y.npc = "top", size=2) +
  facet_grid(nutrient~samplingTime, scale='free') +
  theme_classic()+
  scale_y_log10()+
  scale_x_log10()+
  xlab("Rep3")+
  ylab("Rep2")+
  ggtitle("Scatter plot for Rep1 vs Rep2 (Normalized ratio)")
```

### Combine ratios and intensity into one table 
```{r, echo = FALSE, warning = FALSE, message=FALSE}
master <- left_join(intensity_geoMean_normed %>% select(-channel),ratio_all%>% select(-channel)) %>%
  melt() %>%
  rename(data = variable)

dim(master)
head(master)
write_csv(master, "./tables/master_protein.csv")
```

## PCA analysis for each data type
```{r, echo = FALSE, warning = FALSE, message=FALSE, fig.height=5, fig.width=8}
#heatmap of core data
##specify the component you are interested in 
n = 1
m = 2
for (i in c("intensity","normed_intensity","ratio","normed_ratio")) {
  df <- master %>% 
    #remove spike-in in the data frame
    filter(genotype != "spike-in") %>%
    #remove samples from nitrogen starved condition
    filter(nutrient != "N") %>%
    #filter(replicate != "R2") %>%
    filter(data == i) %>%
    select(`Protein IDs`, `Gene names`, genotype, nutrient, replicate, samplingTime, value) %>%
    filter(!is.na(value)) %>%
    filter(is.finite(value)) %>%
    filter(value > 0 ) %>%
    unite(cond, nutrient, genotype, samplingTime, replicate) %>%
    spread(key = cond, value = value) 

  annotation_col <- names(df)[-c(1:2)] %>% as.tibble() %>%
    separate(value, c("nutrient","genotype","samplingTime","rep"), sep = "_", remove = FALSE) %>%
    column_to_rownames(var = "value")

  breaksList = seq(-0.07, 0.07, by = 0.0001)
#make heatmaps
  pheatmap(log2(df[,-c(1:2)]) %>% drop_na(), 
           colorRampPalette(rev(brewer.pal(n =10, name = "RdYlBu")))(length(breaksList)),
           cluster_cols = T, scale = "row", 
           annotation_col = annotation_col, main = paste0("Log2(",i,")"))
  
  #the column number here is determined by the conditions + replicates + mutant libraries
     # dz <- data.frame(t(log2(df[,-c(1:2)]) %>% drop_na()))
      dz <- data.frame(t(df[,-c(1:2)] %>% drop_na()))
      dz[,ncol(dz)+1] <- rownames(dz)
      colnames(dz)[ncol(dz)] <- "condition"
  
      dz <- dz %>% 
        separate(condition, c("nutrient", "genotype",  "samplingTime", "rep"), sep = "_", remove = FALSE) %>%
        unite(cond, nutrient, genotype, samplingTime, remove = FALSE) 
      dz$genotype <- factor(dz$genotype, levels = c("WT","rim15KO"))
  
      dim(dz)
      l<-length(dz)-6
      
      #plot wt in C
      
      tmp <- dz %>% filter(genotype == "WT" & nutrient == "C")
      #autoplot(pam(tmp[,c(1:l)], 4), frame = TRUE, frame.type = 'norm')
      a <- autoplot(prcomp(tmp[,c(1:l)], scale = T), x = n,    # PC2
         y = m, data = tmp, colour = 'samplingTime', shape = "rep",size = 4) +
              theme_classic(base_size = 12) +
              theme( legend.position = "bottom") +
              #facet_wrap(genotype~nutrient, scale = "free")+
              scale_color_manual(values=c("#f7c53b","#54c922","#2fcefa" ,"#225fc9")) +#,"#9dcc5f", "#0d5f8a")) +
              ggtitle("WT_C")+
              scale_fill_manual(values=c("#f7c53b","#54c922","#2fcefa" ,"#225fc9")) #"#bfbdbd",
      
      tmp <- dz %>% filter(genotype == "WT" & nutrient == "P")
      b <- autoplot(prcomp(tmp[,c(1:l)], scale = T), x = n,    # PC2
         y = m, data = tmp, colour = 'samplingTime', shape = "rep",size =4) +
              theme_classic(base_size = 12) +
              theme(legend.position = "bottom") +
              #facet_wrap(genotype~nutrient, scale = "free")+
              scale_color_manual(values=c("#f7c53b","#54c922","#2fcefa" ,"#225fc9")) +#,"#9dcc5f", "#0d5f8a")) +
              ggtitle("WT_P")+
              scale_fill_manual(values=c("#f7c53b","#54c922","#2fcefa" ,"#225fc9")) #"#bfbdbd",
      tmp <- dz %>% filter(genotype == "rim15KO" & nutrient == "C")
      c <- autoplot(prcomp(tmp[,c(1:l)], scale = T), x = n,    # PC2
         y = m, data = tmp, colour = 'samplingTime', shape = "rep",size = 4) +
              theme_classic(base_size = 12) +
              theme(legend.position = "bottom") +
              #facet_wrap(genotype~nutrient, scale = "free")+
              scale_color_manual(values=c("#f7c53b","#54c922","#2fcefa" ,"#225fc9")) +#,"#9dcc5f", "#0d5f8a")) +
              ggtitle("rim15_C")+
              scale_fill_manual(values=c("#f7c53b","#54c922","#2fcefa" ,"#225fc9")) #"#bfbdbd",
      tmp <- dz %>% filter(genotype == "rim15KO" & nutrient == "P")
      d <- autoplot(prcomp(tmp[,c(1:l)], scale = T), x = n,    # PC2
         y = m, data = tmp, colour = 'samplingTime', shape = "rep",size = 4) +
              theme_classic(base_size = 12) +
              theme(legend.position = "bottom") +
              #facet_wrap(genotype~nutrient, scale = "free")+
              scale_color_manual(values=c("#f7c53b","#54c922","#2fcefa" ,"#225fc9")) +#,"#9dcc5f", "#0d5f8a")) +
              ggtitle("rim15_P")+
              scale_fill_manual(values=c("#f7c53b","#54c922","#2fcefa" ,"#225fc9")) #"#bfbdbd",
      print(ggarrange(a, b, c, d, ncol = 2, nrow = 2))     # First row with scatter plot
      ggarrange(a, b, c, d, ncol = 2, nrow = 2) # Second row with box and dot plots
      ggsave(paste0("./Figure 3/Fig3A",i,".pdf"), width = 8, height = 7)
  
  ##pca for each timepoint btw genotypes
           #plot wt in C
    for (j in c("C","P")){
      tmp <- dz %>% filter(nutrient == j & samplingTime == "T00")
      a <- autoplot(prcomp(tmp[,c(1:l)], scale = T), data = tmp, colour = 'genotype', shape = "rep",size = 4) +
              theme_classic(base_size = 12) +
              theme(legend.position = "bottom") +
              #facet_wrap(genotype~nutrient, scale = "free")+
              scale_color_manual(values=c("#00AFBB","#E7B800")) +#,"#9dcc5f", "#0d5f8a")) +
              ggtitle(paste0("T00_",j))+
              scale_fill_manual(values=c("#00AFBB","#E7B800")) #"#bfbdbd",
      tmp <- dz %>% filter(nutrient == j & samplingTime == "T06")
      b <- autoplot(prcomp(tmp[,c(1:l)], scale = T), data = tmp, colour = 'genotype', shape = "rep",size = 4) +
              theme_classic(base_size = 12) +
              theme(legend.position = "bottom") +
              #facet_wrap(genotype~nutrient, scale = "free")+
              scale_color_manual(values=c("#00AFBB","#E7B800")) +#,"#9dcc5f", "#0d5f8a")) +
              ggtitle(paste0("T06_",j))+
              scale_fill_manual(values=c("#00AFBB","#E7B800")) #"#bfbdbd",
      tmp <- dz %>% filter(nutrient == j & samplingTime == "T16")
      c <- autoplot(prcomp(tmp[,c(1:l)], scale = T), data = tmp, colour = 'genotype', shape = "rep",size = 4) +
              theme_classic(base_size = 12) +
              theme(legend.position = "bottom") +
              #facet_wrap(genotype~nutrient, scale = "free")+
              scale_color_manual(values=c("#00AFBB","#E7B800")) +#,"#9dcc5f", "#0d5f8a")) +
              ggtitle(paste0("T16_",j))+
              scale_fill_manual(values=c("#00AFBB","#E7B800")) #"#bfbdbd",
      tmp <- dz %>% filter(nutrient == j & samplingTime == "T30")
      d <- autoplot(prcomp(tmp[,c(1:l)], scale = T), data = tmp, colour = 'genotype', shape = "rep",size = 4) +
              theme_classic(base_size = 12) +
              theme(legend.position = "bottom") +
              #facet_wrap(genotype~nutrient, scale = "free")+
              scale_color_manual(values=c("#00AFBB","#E7B800")) +#,"#9dcc5f", "#0d5f8a")) +
              ggtitle(paste0("T30_",j))+
              scale_fill_manual(values=c("#00AFBB","#E7B800")) #"#bfbdbd",
      print(ggarrange(a, b, c, d, ncol = 4)) # Second row with box and dot plots
      ggarrange(a, b, c, d, ncol = 4)
      ggsave(paste0("./Figure 3/Fig3SA",i,"_",j,".pdf"), width = 8, height = 5)
    }
  }
```

#### clearly, normalized ratio maximized the difference on proteome proflies between genotypes

## Filter out the poorly correlated replicates
#### 1. WT -C T06 rep1, 
#### 2. WT -C T16 rep1, 
#### 3. WT -C T30 rep2, 
#### 4. rim15 -C T16 rep2, 
#### 5. rim15 -C T30 rep2, 
#### 6. WT -P T16 rep1, 
#### 7. rim15KO -P T16 rep1
```{r}
master <- read_csv("./tables/master_protein.csv")

filtered <- master %>% 
    #remove spike-in in the data frame
  filter(genotype != "spike-in") %>%
    #remove samples from nitrogen starved condition
  filter(nutrient != "N") %>%
    #filter(replicate != "R2") %>%
  filter(data == "normed_ratio") %>% 
  melt() %>%
  dplyr::select(-variable) %>%
  #remove na values
  dplyr::filter(!is.na(value)) %>% 
  #remove infinite values
  dplyr::filter(is.finite(value)) %>% 
  #remove 0 values, wich will be a problem for later log based transformation
  dplyr::filter(value > 0) %>%
  group_by(`Protein IDs`,`Gene names`, nutrient, samplingTime, replicate, data) %>%
  mutate(genotype_number = n()) %>%
  filter(genotype_number == 2) %>%
  group_by(`Protein IDs`,`Gene names`, nutrient, samplingTime, genotype, data) %>%
  mutate(rep_numner = n()) %>%
  filter(rep_numner == 3) %>%
  group_by(`Protein IDs`,`Gene names`, nutrient, replicate, genotype, data) %>%
  mutate(time_number = n()) %>%
  filter(time_number == 4) %>%
  filter(!(
    #1. WT -C T06 rep1, 
    (genotype == 'WT' & nutrient == "C" & samplingTime == 'T06' & replicate == 'R1') |
      #WT -C T16 rep1
      (genotype == 'WT' & nutrient == "C" & samplingTime == 'T16' & replicate == 'R1') |
      #WT -C T30 rep2, 
      (genotype == 'WT' & nutrient == "C" & samplingTime == 'T30' & replicate == 'R2') |
      #rim15 -C T16 rep2, 
      (genotype == 'rim15KO' & nutrient == "C" & samplingTime == 'T16' & replicate == 'R2') |
      #rim15 -C T30 rep2, 
      (genotype == 'rim15KO' & nutrient == "C" & samplingTime == '30' & replicate == 'R2') |
      #6. WT -P T16 rep1, 
      (genotype == 'WT' & nutrient == "P" & samplingTime == 'T16' & replicate == 'R1') |
      #7. rim15KO -P T16 rep1
      (genotype == 'rim15KO' & nutrient == "P" & samplingTime == 'T16' & replicate == 'R1') ))

#save
write_csv (filtered, "./tables/filtered_norm_ratios_WP.csv")
```

## T-test for DE at each timepoint 
```{r, echo = FALSE, warning = FALSE, message=FALSE}
#drop na function
delete.na <- function(DF, n=0) {DF[rowSums(is.na(DF)) <= n,]}

#t-test between genotypes 
pG_reshape <- master %>% 
  filter(genotype != "spike-in") %>%
  filter(nutrient != "N") %>%
  #filter(replicate != "R2") %>%
  group_by(`Protein IDs`, `Protein names`,`Gene names`,`Fasta headers`, data, nutrient, samplingTime) %>%
  unite(cond, genotype, replicate) %>%
  spread(key = cond, value = value) 

#remove the rows whoes has more than 2 NA values
pG_filtered <- delete.na(pG_reshape,2)
  
#run t-test and log2FC between genotypes for each timepoint
ttest_pG <- pG_filtered %>%
  melt() %>%
  separate(variable, c("genotype","replicate")) %>%
  group_by(`Protein IDs`, `Protein names`,`Gene names`,`Fasta headers`, nutrient, samplingTime, data) %>%
  summarise_each(funs(t.test(.[genotype == "WT"], .[genotype == "rim15KO"], var.equal=TRUE)$p.value), pvalue = value) 

ttest_pG$qvalue <- p.adjust(ttest_pG$pvalue,method="BH") 

log2FC_pG <- pG_filtered %>%
  melt() %>%
  separate(variable, c("genotype","replicate")) %>%
  group_by(`Protein IDs`, `Protein names`,`Gene names`,`Fasta headers`, nutrient, samplingTime, data) %>%
  summarise_each(funs(mean(.[genotype=="WT"], na.rm=TRUE)/mean(.[genotype == "rim15KO"], na.rm=TRUE)), meanDF = value) %>%
  mutate(log2FC = log2(meanDF))

logFC_ttest_pG <-left_join(ttest_pG, log2FC_pG) %>% 
  #remove na values in log2FC
  filter(!is.na(meanDF)) 

write_csv(logFC_ttest_pG,"./tables/DE_proteins_byGenotype.csv")

carbon <- logFC_ttest_pG %>% filter(nutrient == "C") 

ggplot(carbon, aes(x=log2FC, y =-log10(qvalue))) +
  geom_point(data = filter(carbon, qvalue > 0.1), alpha = 1/6, size = 0.8, col = "grey") +
  geom_point(data = filter(carbon, log2FC > 0, qvalue < 0.1), alpha = 1/2, size = 1.5, col = "yellow") +
  geom_point(data = filter(carbon, log2FC < 0, qvalue < 0.1), alpha = 1/2, size = 1.5, col = "blue") +
  theme_classic() +
 # coord_fixed(ratio = 6) +
  xlab(expression(Log[2]*" (WT / rim15Δ)")) +
  ylab(expression(-Log[10]* "(qvalue)")) +
  #xlim(-2.5,6.5) +
  #ylim(0,2) +
      ggtitle("carbon starvation") +
  theme(plot.title = element_text(hjust = 0.5, size = 14), axis.title = element_text(size = 12), axis.text = element_text(size = 10, colour = "black")) +
  facet_grid(samplingTime~data, scale = "free")+
  theme(legend.position="none")
ggsave("./plots/protein_volcano_carbon.pdf",width = 7, height = 4,  useDingbats=FALSE)

Pi <- logFC_ttest_pG %>% filter(nutrient == "P") 

ggplot(Pi, aes(x=log2FC, y =-log10(qvalue))) +
  geom_point(data = filter(Pi, qvalue > 0.1), alpha = 1/6, size = 0.8, col = "grey") +
  geom_point(data = filter(Pi, log2FC > 0, qvalue < 0.1), alpha = 1/2, size = 1.5, col = "#e7b800") +
  geom_point(data = filter(Pi, log2FC < 0, qvalue < 0.1), alpha = 1/2, size = 1.5, col = "#00afbb") +
  theme_classic() +
 # coord_fixed(ratio = 6) +
  xlab(expression(Log[2]*" (WT / rim15Δ)")) +
  ylab(expression(-Log[10]* "(qvalue)")) +
    ggtitle("phosphorous starvation") +
  #xlim(-2.5,6.5) +
  #ylim(0,2) +
  theme(plot.title = element_text(hjust = 0.5, size = 14), axis.title = element_text(size = 12), axis.text = element_text(size = 10, colour = "black")) +
  facet_grid(samplingTime~data, scale = "free")+
  theme(legend.position="none")
ggsave("./plots/protein_volcano_phosphorous.pdf",width = 7, height = 4,  useDingbats=FALSE)

#t.test for DE protein between nutrient 
pG_reshape <- master %>% 
  group_by(`Protein IDs`, `Protein names`,`Gene names`,`Fasta headers`, data, nutrient, samplingTime) %>%
  unite(cond, nutrient, replicate) %>%
  spread(key = cond, value = value) 

#remove the rows whoes has more than 2 NA values
pG_filtered <- delete.na(pG_reshape,2)

ttest_pG <- pG_filtered %>%
  melt() %>%
  separate(variable, c("nutrient","replicate")) %>%
  group_by(`Protein IDs`, `Protein names`,`Gene names`,`Fasta headers`, genotype, samplingTime, data) %>%
  summarise_each(funs(t.test(.[nutrient == "C"], .[nutrient == "P"], var.equal=TRUE)$p.value), pvalue = value) 

ttest_pG$qvalue <- p.adjust(ttest_pG$pvalue,method="BH") 

log2FC_pG <- pG_filtered %>%
  melt() %>%
  separate(variable, c("nutrient","replicate")) %>%
  group_by(`Protein IDs`, `Protein names`,`Gene names`,`Fasta headers`, genotype, samplingTime, data) %>%
  summarise_each(funs(mean(.[nutrient=="C"], na.rm=TRUE)/mean(.[nutrient == "P"], na.rm=TRUE)), meanDF = value) %>%
  mutate(log2FC = log2(meanDF))

logFC_ttest_pG <-left_join(ttest_pG, log2FC_pG) %>% 
  #remove na values in log2FC
  filter(!is.na(meanDF)) 

write_csv(logFC_ttest_pG,"./tables/DE_proteins_byNutrient.csv")

wt <- logFC_ttest_pG %>% filter(genotype == "WT") 

ggplot(wt, aes(x=log2FC, y =-log10(qvalue))) +
  geom_point(data = filter(wt, qvalue > 0.05), alpha = 1/6, size = 0.8, col = "grey") +
  geom_point(data = filter(wt, log2FC > 0, qvalue < 0.05), alpha = 1/2, size = 1.5, col = "#e7b800") +
  geom_point(data = filter(wt, log2FC < 0, qvalue < 0.05), alpha = 1.2, size = 1.5, col = "#00afbb") +
  theme_classic() +
 # coord_fixed(ratio = 6) +
  xlab(expression(Log[2]*" (carbon / phosphorous)")) +
  ylab(expression(-Log[10]* "(qvalue)")) +
  #xlim(-2.5,6.5) +
  #ylim(0,2) +
    ggtitle("wt") +
  theme(plot.title = element_text(hjust = 0.5, size = 14), axis.title = element_text(size = 12), axis.text = element_text(size = 10, colour = "black")) +
  facet_grid(samplingTime~data, scale = "free")+
  theme(legend.position="none")
ggsave("./plots/protein_volcano_wt.pdf",width = 7, height = 4,  useDingbats=FALSE)

rim15 <- logFC_ttest_pG %>% filter(genotype == "rim15KO") 

ggplot(rim15, aes(x=log2FC, y =-log10(qvalue))) +
  geom_point(alpha = 1/6, size = 0.8, col = "grey") +
  geom_point(data = filter(rim15, qvalue > 0.05), alpha = 1/6, size = 0.8, col = "grey") +
  geom_point(data = filter(rim15, log2FC > 1, qvalue < 0.05), alpha = 1/2, size = 1.5, col = "#e7b800") +
  geom_point(data = filter(rim15, log2FC < -1, qvalue < 0.05), alpha = 1/2, size = 1.5, col = "#00afbb") +
  theme_classic() +
 # coord_fixed(ratio = 6) +
  xlab(expression(Log[2]*" (carbon / phosphorous)")) +
  ylab(expression(-Log[10]* "(qvalue)")) +
  #xlim(-2.5,6.5) +
  #ylim(0,2) +
  ggtitle("rim15") +
  theme(plot.title = element_text(hjust = 0.5, size = 14), axis.title = element_text(size = 12), axis.text = element_text(size = 10, colour = "black")) +
  facet_grid(samplingTime~data, scale = "free")+
  theme(legend.position="none")
ggsave("./plots/protein_volcano_rim15.pdf",width = 7, height = 4,  useDingbats=FALSE)
```