---
title: "R Notebook"
output: html_notebook
---

Combined protein and pSites analysis
##merge protein Groups with pSite
```{r , echo = FALSE}
logFC_ttest <- left_join(logFC_ttest_pS, logFC_ttest_pG, by = c("Protein names","Gene names","Fasta headers","nutrient","time"), suffix = c(".phospho",".protein"))

ggplot(data=logFC_ttest, aes(x=log2FC.protein, y =log2FC.phospho)) +
  #geom_point(data=logFC_ttest, alpha = 1/3, size = 0.6, color = "#6e6e6e") +
  geom_point(data=logFC_ttest, size=1, alpha = 0.7, col="#a3a3a3") +
  geom_point(data=logFC_ttest %>% dplyr::filter(log2FC.phospho > 0 & qvalue.phospho < 0.05 & log2FC.phospho > log2FC.protein), color='#FF3333',size=1, alpha = 0.7) +
  geom_point(data=logFC_ttest %>% dplyr::filter(log2FC.phospho < 0 & qvalue.phospho < 0.05  & log2FC.phospho < log2FC.protein), color='#3399FF',size=1, alpha = 0.7) +
  #stat_cor(method = "spearman", label.x.npc = "left", label.y.npc = "top", size=3, col = "#a3a3a3") +
  theme_classic() +
 geom_text_repel(data = logFC_ttest %>% dplyr::filter(log2FC.phospho > 0 & qvalue.phospho < 0.05 & log2FC.phospho > log2FC.protein), aes(label = pSite), size =2.5) +
 geom_text_repel(data = logFC_ttest %>% dplyr::filter(log2FC.phospho < 0 & qvalue.phospho < 0.05  & log2FC.phospho < log2FC.protein), aes(label = pSite), size =2.5) +
  #geom_vline(xintercept = 0, color = "#949191")+
  #geom_hline(yintercept = 0, color = "#949191") +
  xlab(expression(log[2]*" (WT / rim15Δ) - protein")) +
  ylab(expression(log[2]*" (WT / rim15Δ) - phosphorylation site")) +
  theme(plot.title = element_text(hjust = 0.5, size = 14), axis.title = element_text(size = 12), axis.text = element_text(size = 10, colour = "black"))+
  #xlim(c(-3, 3)) +
  facet_grid(nutrient~time, scale = "free")+
  theme(legend.position="none")
```

#significantly differentially expressing proteins

#Significantly DE proteins by genotype
```{r, eval=FALSE}
protein_list <- unique(sig_genotype$`Protein IDs`)
plot_list_genotype = list()

for (cond in c("C","P")) {
  for (i in 1:length(protein_list)) {
    protein <- master %>% 
      filter(nutrient == cond) %>%
      filter(`Protein IDs` %in% sig_genotype$`Protein IDs`) %>%
      melt() %>%
      mutate(samplingTime = substr(samplingTime, 2, nchar(samplingTime))) %>% 
      mutate(samplingTime = as.numeric(samplingTime)) %>% 
      filter(`Protein IDs` == protein_list[i]) 
    
    p <- protein %>%
      ggline(x = "samplingTime", y = "value", color = "genotype", facet.by = "variable",  add = c("mean_se", "dotplot"), palette = c("#00AFBB","#E7B800"))+
      facet_wrap( ~ variable, scale = "free", ncol = 4) +
      ggtitle(paste0(cond, " starvation ", unique(protein$`Gene names`))) +
      scale_y_log10()
 
  plot_list_genotype[[i]] = p
  }
}

ggsave(filename = "./plots/sigDE_by_genotype.pdf", plot = marrangeGrob(plot_list_genotype, nrow=6, ncol = 1), width = 15, height = 10)
```

#Significantly DE proteins by nutrient
```{r, echo = FALSE}
protein_list <- unique(sig_nutrient$`Protein IDs`)
plot_list_nutrient = list()

for (cond in c("rim15KO","WT")) {
  for (i in 1:length(protein_list)) {
    protein <- master %>% 
      filter(`Protein IDs` %in% sig_nutrient$`Protein IDs`) %>%
      filter(nutrient == cond) %>%
      melt() %>%
      mutate(samplingTime = substr(samplingTime, 2, nchar(samplingTime))) %>% 
      mutate(samplingTime = as.numeric(samplingTime)) %>% 
      filter(`Protein IDs` == protein_list[i]) 
    
    p <- protein %>%
      ggline(x = "samplingTime", y = "value", color = "nutrient", facet.by = "variable",  add = c("mean_se", "dotplot"), palette = c("#ff9500","#8300ab"))+
      facet_wrap( ~ variable, scale = "free", ncol = 4) +
      ggtitle(paste0(cond, " starvation ", unique(protein$`Gene names`))) +
      scale_y_log10()
 
  plot_list_nutrient[[i]] = p
  }
}

ggsave(filename = "./plots/sigDE_by_nutrient.pdf", plot = marrangeGrob(plot_list_nutrient, nrow=6, ncol = 1), width = 15, height = 10)
```

#Significantly DE proteins by nutrient
```{r, echo = FALSE}
protein_list <- unique(sig_3way$`Protein IDs`)
plot_list_3way = list()

for (cond in c("rim15KO","WT")) {
  for (i in 1:length(protein_list)) {
    protein <- master %>% 
      filter(`Protein IDs` %in% sig_3way$`Protein IDs`) %>%
      filter(nutrient == cond) %>%
      melt() %>%
      mutate(samplingTime = substr(samplingTime, 2, nchar(samplingTime))) %>% 
      mutate(samplingTime = as.numeric(samplingTime)) %>% 
      filter(`Protein IDs` == protein_list[i]) 
    
    p <- protein %>%
      ggline(x = "samplingTime", y = "value", color = "nutrient", facet.by = "variable",  add = c("mean_se", "dotplot"), palette = c("#ff9500","#8300ab"))+
      facet_wrap( ~ variable, scale = "free", ncol = 4) +
      ggtitle(paste0(cond, " starvation ", unique(protein$`Gene names`))) +
      scale_y_log10()
 
  plot_list_3way[[i]] = p
  }
}

ggsave(filename = "./plots/sigDE_3way.pdf", plot = marrangeGrob(plot_list_3way, nrow=6, ncol = 1), width = 15, height = 10)
```

## Phosphoproteomic data preperation
```{r , echo = FALSE}

```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
