---
title: "phospho proteomics analysis (based on normalized ratio)"
author: "Siyu Sun"
date: "5/4/2020"
output: 
  html_document:
    keep_md: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(collapse=TRUE, comment="##", fig.retina=2, fig.path = "README_figs/README-")
```

#load packages
```{r, include=FALSE}
library(dplyr)
library(ggplot2)
library(reshape2)
library(tidyverse)
library(GGally)
library(ggpubr)
library(ggrepel)
library(gridExtra)
library(devtools)
library(ggfortify)
library(limma)
library(plotly)
library(broom)
library(rmotifx)
library(pheatmap)
```

#read in pSite table
```{r, include=FALSE}
#read in pSite groups 
phosphosites <- read_csv("./data/Phospho (STY)Sites.csv")
#View(phosphosites)
```

#look at the pSite normalized ratios
```{r, echo = FALSE}
#extract the protome info
ratio_normed <- phosphosites %>%
  dplyr::select(Proteins, id, `Positions within proteins`,`Leading proteins`,`Protein names`,`Gene names`,`Fasta headers`, `Localization prob`,`Sequence window`, contains("Ratio")) %>%
  dplyr::select(Proteins, id, `Sequence window`, `Positions within proteins`,`Leading proteins`,`Protein names`,`Gene names`,`Fasta headers`, `Localization prob`, contains("phospho")) %>%
  dplyr::select(Proteins, id, `Positions within proteins`,`Leading proteins`,`Protein names`,`Gene names`,`Fasta headers`, `Localization prob`,`Sequence window`, contains("normalized")) %>%
  dplyr::select(-matches("_N_")) %>%
  dplyr::select(-matches("___")) %>%
  mutate(id = factor(id)) %>%
  unite(pSite, `Gene names`, `Positions within proteins`, remove = FALSE) %>%
  dplyr::filter(`Localization prob` > 0.7) %>%
  select(-c(`Localization prob`, `Positions within proteins`)) %>%
  dplyr::select(-contains("variability")) %>%
  dplyr::select(-contains("count")) %>%
  dplyr::select(-contains("iso-count")) %>%
  dplyr::select(-contains("type"))

#extract data for time 0
T00_P <- ratio_normed %>% dplyr::select(Proteins, id, pSite, `Leading proteins`,`Protein names`,`Gene names`,`Fasta headers`, contains("T00")) %>% 
  dplyr::rename(`Ratio M/L normalized phospho_P_T00_R1` = `Ratio M/L normalized phospho_C_T00_R1`,
         `Ratio M/L normalized phospho_P_T00_R2` = `Ratio M/L normalized phospho_C_T00_R2`,
         `Ratio M/L normalized phospho_P_T00_R3` = `Ratio M/L normalized phospho_C_T00_R3`,
         `Ratio H/L normalized phospho_P_T00_R1` = `Ratio H/L normalized phospho_C_T00_R1`,
         `Ratio H/L normalized phospho_P_T00_R2` = `Ratio H/L normalized phospho_C_T00_R2`,
         `Ratio H/L normalized phospho_P_T00_R3` = `Ratio H/L normalized phospho_C_T00_R3`,
         `Ratio H/M normalized phospho_P_T00_R1` = `Ratio H/M normalized phospho_C_T00_R1`,
         `Ratio H/M normalized phospho_P_T00_R2` = `Ratio H/M normalized phospho_C_T00_R2`,
         `Ratio H/M normalized phospho_P_T00_R3` = `Ratio H/M normalized phospho_C_T00_R3`)

ratio_normed_add<- dplyr::inner_join(ratio_normed, T00_P, by = c("Proteins","id", "pSite", "Leading proteins", "Gene names","Protein names","Fasta headers"))

#reshape the table into a tidy format 
ratio_normed_tidy <- melt(ratio_normed_add) %>%
  separate(variable, c("signal","channel","normed","sampleNum"), sep = " ") %>%
  #separate(sampleNum, c("cond","site"), sep = "___") %>%
  #filter(variable %in% c())
  separate(sampleNum, c("omic","nutrient","samplingTime","replicate"), sep = "_") %>%
  dplyr::select(-c("signal","omic","normed"))  %>%
  #select only wild type and ko ratios individually
  dplyr::filter(channel %in% c("H/L","M/L","H/M"))

#box plot
ggplot(ratio_normed_tidy, aes(x=samplingTime, y=value, fill = channel)) + 
  geom_boxplot(position=position_dodge(0.8))+#, outlier.shape = NA) +
  facet_grid(nutrient~replicate) +
  theme_classic() +
  scale_color_manual(values=c("#13b1f0","#e7c100","#eb5534"))+
  scale_fill_manual(values=c("#13b1f0","#e7c100","#eb5534"))+
  scale_y_log10() +
  ylab("Ratio")

#summary table
summary_table <- ratio_normed_tidy %>% 
  group_by (channel, nutrient, samplingTime, replicate) %>%
  dplyr::summarize (total = sum (na.omit(value)), mean = mean(na.omit(value)), median = median(na.omit(value)), entries = n())

non_NAs <- ratio_normed_tidy %>% 
  drop_na() %>%
  group_by (channel, nutrient, samplingTime, replicate) %>%
  dplyr::summarize (total = sum (na.omit(value)), mean = mean(na.omit(value)), median = median(na.omit(value)), entries = n())

non_NAs %>%
  #remove values that is 0
  ggplot(aes(x=samplingTime,y = entries, col = replicate, fill = replicate, group = replicate)) + 
  geom_point(aes(x=samplingTime,y = entries, col = replicate, fill = replicate))+
  geom_line(aes(x=samplingTime,y = entries, col = replicate, fill = replicate))+
  theme(axis.text.x=element_text(angle=0),legend.position="bottom")+
  facet_grid(channel ~ nutrient, scales = "free") +
  scale_color_manual(values=c("#789cff", "#656371", "#ff7369"))+
  scale_fill_manual(values=c("#789cff", "#656371", "#ff7369")) +
  theme_classic() +
  xlab("Hours post nutrient depletion") +
  ylab("unique pSite identified")

#look at the distribution of proteins 
ratio_normed_tidy %>%
  #remove values that is 0
  ggplot(aes(x=log2(value), col = replicate, fill = replicate)) + 
  geom_histogram(aes(y=..density..),  position="identity", alpha=0.1)+
  geom_density(alpha = 0.3)+
  theme(axis.text.x=element_text(angle=0),legend.position="bottom")+
  facet_grid(channel ~ nutrient + samplingTime, scales = "free") +
  scale_color_manual(values=c("#789cff", "#656371", "#ff7369"))+
  scale_fill_manual(values=c("#789cff", "#656371", "#ff7369")) +
  theme_classic() +
  ggtitle("Ratio distribution")

#coorelation Among Replicates 
df_spread <- ratio_normed_tidy %>% 
  spread(key = replicate, value = value) 
  
ggscatter(df_spread, x = "R1", y = "R2", alpha = 0.3, merge = TRUE, size=0.8,  col= "channel", shape = "channel", palette = c("#50a3a4", "#fcaf38", "#f95335")) + # Add confidence interval)
  stat_cor(aes(color = channel), method = "pearson", label.x.npc = "left", label.y.npc = "top", size=2) +
  facet_grid(nutrient~samplingTime, scale='free') +
  theme_classic()+
  scale_y_log10()+
  scale_x_log10()+
  xlab("Rep1")+
  ylab("Rep2")+
  ggtitle("Scatter plot for Rep1 vs Rep2 (build-in pSite ratio)")

  #coorelation between rep1 and rep3
ggscatter(df_spread, x = "R1", y = "R3", alpha = .3, size=0.8, cor.coef.size = 0.4,  col= "channel", shape = "channel", palette = c("#50a3a4", "#fcaf38", "#f95335")) + # Add confidence interval)
    stat_cor(aes(color = channel), method = "pearson", label.x.npc = "left", label.y.npc = "top", size=2)  +
    facet_grid(nutrient~samplingTime, scale='free') +
        theme_classic()+
    scale_y_log10()+
    scale_x_log10()+
    ylab("Rep3")+
    xlab("Rep1")+
    ggtitle("Scatter plot for Rep1 vs Rep3 (build-in pSite ratio)")

##reversed build-in ratio by genotypes
rep2_HL_ML <- ratio_normed_tidy %>% 
  dplyr::filter(channel == "H/L" & replicate == "R2" & nutrient == "P") %>%
  mutate(channel="M/L")
##swith M/L to H/L
rep2_ML_HL <- ratio_normed_tidy %>% 
  dplyr::filter(channel == "M/L" & replicate == "R2" & nutrient == "P") %>%
  mutate(channel="H/L")
rep2_MH_MH <-  ratio_normed_tidy %>% 
  dplyr::filter(channel == "H/M" & replicate == "R2" & nutrient == "P") %>%
  mutate(value=1/value)
##combine all new replicate 2 data
rep2 <- bind_rows(rep2_HL_ML, rep2_ML_HL,rep2_MH_MH)

#join the revised replicate 2 with the other data from replicate 1&3
#remove old data
new_re <- anti_join(ratio_normed_tidy, ratio_normed_tidy %>% dplyr::filter (nutrient == "P" & replicate == "R2"))

ratio_normed_tidy_re <- bind_rows(rep2, new_re) %>%
  mutate(channel=replace(channel, channel=="M/L", "rim15KO")) %>%
  mutate(channel=replace(channel, channel=="H/L", "WT")) %>%
  mutate(channel=replace(channel, channel=="H/M", "WT/rim15KO")) %>%
  dplyr::rename(genotype = channel) %>%
  #replace NA values with 1, which assuming there is no changes
  mutate(value = if_else(is.na(value), 1, value))

#box plot
ggplot(ratio_normed_tidy_re, aes(x=genotype, y=log2(value), fill = samplingTime)) + 
  geom_boxplot(position=position_dodge(0.8))+#, outlier.shape = NA) +
  facet_grid(nutrient~replicate) +
  theme_classic() +
    theme(axis.text.x=element_text(angle=0),legend.position="bottom")+
  #scale_color_manual(values=c("#13b1f0","#e7c100","#eb5534"))+
  #scale_fill_manual(values=c("#13b1f0","#e7c100","#eb5534"))+
  ylab("pSites Ratio")

#box plot for different genotypes at T00
ggplot(ratio_normed_tidy_re %>% dplyr::filter(samplingTime == "T00"), aes(x=genotype, y=log2(value), fill = samplingTime)) + 
  geom_boxplot(position=position_dodge(0.8), outlier.shape = NA) +
  facet_grid(nutrient~replicate) +
  theme_classic() +
  theme(axis.text.x=element_text(angle=0),legend.position="bottom")+
  #scale_color_manual(values=c("#13b1f0","#e7c100","#eb5534"))+
  #scale_fill_manual(values=c("#13b1f0","#e7c100","#eb5534"))+
  ylab("log2(pSites Ratio)")

#look at the distribution of proteins 
ratio_normed_tidy_re %>%
  #remove values that is 0
  ggplot(aes(x=log2(value), col = replicate, fill = replicate)) + 
  geom_histogram(aes(y=..density..),  position="identity", alpha=0.1)+
  geom_density(alpha = 0.3)+
  theme(axis.text.x=element_text(angle=0),legend.position="bottom")+
  facet_grid(genotype ~ nutrient + samplingTime, scales = "free") +
  scale_color_manual(values=c("#789cff", "#656371", "#ff7369"))+
  scale_fill_manual(values=c("#789cff", "#656371", "#ff7369")) +
  theme_classic() +
  ggtitle("Scatter plot for Rep1 vs Rep3 (build-in pSites ratio)")

#coorelation Among Replicates 
df_spread <- ratio_normed_tidy_re %>% 
  spread(key = replicate, value = value) 
  
ggscatter(df_spread, x = "R1", y = "R2", alpha = 0.3, merge = TRUE, size=0.8,  col= "genotype", shape = "genotype", palette = c("#50a3a4", "#fcaf38", "#f95335")) + # Add confidence interval)
  stat_cor(aes(color = genotype), method = "pearson", label.x.npc = "left", label.y.npc = "top", size=2) +
  facet_grid(nutrient~samplingTime, scale='free') +
  theme_classic()+
  scale_y_log10()+
  scale_x_log10()+
  xlab("Rep1")+
  ylab("Rep2")+
  ggtitle("Scatter plot for Rep1 vs Rep2 (build-in pSites ratio)")

  #coorelation between rep1 and rep3
  ggscatter(df_spread, x = "R1", y = "R3", alpha = .3, size=0.8, cor.coef.size = 0.4,  col= "genotype", shape = "genotype", palette = c("#50a3a4", "#fcaf38", "#f95335")) + # Add confidence interval)
    stat_cor(aes(color = genotype), method = "pearson", label.x.npc = "left", label.y.npc = "top", size=2)  +
    facet_grid(nutrient~samplingTime, scale='free') +
        theme_classic()+
    scale_y_log10()+
    scale_x_log10()+
    ylab("Rep3")+
    xlab("Rep1")+
    ggtitle("Scatter plot for Rep1 vs Rep3 (build-in pSites ratio)")
  
ratio_normed_core <- ratio_normed_tidy_re  %>%
  #filter(genotype %in% c("rim15KO", "WT")) %>%
  #drop_na("value") %>%
  group_by(pSite, samplingTime, replicate, genotype) %>%
  dplyr::mutate(nutrient_number = n()) %>%
  dplyr::filter(nutrient_number == 2) %>%
  group_by(pSite, nutrient, samplingTime, replicate) %>%
  dplyr::mutate(genotype_number = n()) %>%
  dplyr::filter(genotype_number==3) %>%
  group_by(pSite, nutrient, samplingTime, genotype) %>%
   dplyr::mutate(rep_numner = n()) %>%
  dplyr::filter(rep_numner == 3) %>%
  group_by(pSite, nutrient, replicate, genotype) %>%
   dplyr::mutate(time_number = n()) %>%
  dplyr::filter(time_number == 4)

#heatmap of core data
df <- ratio_normed_core %>% 
  select(nutrient, pSite, genotype, replicate, samplingTime, value) %>%
  unite(cond, nutrient, genotype, samplingTime, replicate) %>%
  spread(key = cond, value = value)

#heatmap
annotation_col <- names(df)[-1] %>% as.tibble() %>%
  separate(value, c("nutrient","genotype","samplingTime","rep"), sep = "_", remove = FALSE) %>%
  column_to_rownames(var = "value")

library(RColorBrewer)
breaksList = seq(-0.07, 0.07, by = 0.0001)

pdf("./plots/pSite_ratio_normed_heat_all.pdf", width = 13, height = 13)
pheatmap(df[,-1] %>% drop_na(), colorRampPalette(rev(brewer.pal(n =10, name = "RdYlBu")))(length(breaksList)),
                   cluster_cols = T, scale = "row", 
                   annotation_col = annotation_col)
dev.off()

pheatmap(df[,-1] %>% drop_na(), colorRampPalette(rev(brewer.pal(n =10, name = "RdYlBu")))(length(breaksList)),
                   cluster_cols = T, scale = "row", 
                   annotation_col = annotation_col)

pdf("./plots/pSite_ratio_normed_heat_light.pdf", width = 10, height = 8)
pheatmap(df[,c(2:25,38:61)] %>% drop_na(), colorRampPalette(rev(brewer.pal(n =10, name = "RdYlBu")))(length(breaksList)),
                   cluster_cols = T, scale = "row", 
                   annotation_col = annotation_col)
dev.off()

pheatmap(df[,c(2:25,38:61)] %>% drop_na(), colorRampPalette(rev(brewer.pal(n =10, name = "RdYlBu")))(length(breaksList)),
                   cluster_cols = T, scale = "row", 
                   annotation_col = annotation_col)

#look at C sample alone
pheatmap(df[,c(2:25)] %>% drop_na(), colorRampPalette(rev(brewer.pal(n =10, name = "RdYlBu")))(length(breaksList)),
                   cluster_cols = T, scale = "row", 
                   annotation_col = annotation_col)

#look at P sample alone
pheatmap(df[,c(38:61)] %>% drop_na(), colorRampPalette(rev(brewer.pal(n =10, name = "RdYlBu")))(length(breaksList)),
                   cluster_cols = T, scale = "row", 
                   annotation_col = annotation_col)
#PCA overtime
#the column number here is determined by the conditions + replicates + mutant libraries
  dz <- data.frame(t(df[,c(2:25,38:61)]%>%drop_na()))
  dz[,ncol(dz)+1] <- rownames(dz)
  colnames(dz)[ncol(dz)] <- "condition"
  dz <- dz %>% 
    separate(condition, c("nutrient", "genotype",  "samplingTime", "rep"), sep = "_", remove = FALSE) %>%
    unite(cond, nutrient, genotype, samplingTime, remove = FALSE)
  dim(dz)
  l<-length(dz)-6
  
pdf("./plots/pSite_ratio_normed_PCA_time.pdf",width=7,height=5)
#the numbers in the range here is subject to the strains left after pre-filtering steps
autoplot(prcomp(dz[,c(1:l)], scale = T), data = dz, colour = 'genotype', size = 3) +
  #theme_classic() +
  theme(legend.position = "bottom") +
  facet_grid(nutrient~samplingTime)+
  scale_color_manual(values=c("#00AFBB","#E7B800"))+#"#6ec3c1" ,"#335120","#9dcc5f", "#0d5f8a")) +
  scale_fill_manual(values=c("#00AFBB","#E7B800")) #"#bfbdbd",
dev.off()

#3D PCA plot
#compute PCA
df<-dz[,c(1:l)]
rownames(df) <- rownames(dz)
res.pca <- prcomp(df, scale = TRUE)

pca <- data.frame(res.pca$x)

plot_ly(x=pca$PC1, y=pca$PC2, z=pca$PC3, type="scatter3d",color=dz$cond, xlab = "PC1", ylab = "PC2", zlab = "PC3")

#reshape the core dataset for perseus 
perseus <- ratio_normed_tidy_re %>% 
  select(Proteins, id, pSite, `Leading proteins`,`Protein names`,`Gene names`,`Fasta headers`, genotype,nutrient,samplingTime,replicate,value) %>%
  unite(cond, genotype, nutrient, samplingTime, replicate) %>%
  spread(key = cond, value = value)
#replace NA value in Genenames column with protein names
perseus$`Gene names` <- ifelse(is.na(perseus$`Gene names`), perseus$`Protein names`, perseus$`Gene names`)  
write_csv(perseus,"./tables/normed_pSite_ratio_perseus.csv")
```

```{r, echo = FALSE}
###############################################################################################
pSite<-read_csv("./tables/normed_pSite_ratio_perseus.csv") %>%
    dplyr::select(-c("id"))

pSite_tidy <- melt(pSite) %>%
  separate(variable, c("genotype", "nutrient", "samplingTime", "replicate"), sep = "_") %>%
  dplyr::filter(genotype %in% c("rim15KO","WT"))

#caculate the profile of each protein
relative_pS <- pSite_tidy %>% unite(cond, genotype, nutrient, replicate) %>%
  drop_na() %>%
  spread(key = samplingTime, value = value) %>%
  mutate(`T00/T00` = T00-T00, `T06/T00` = T06-T00, `T16/T00` = T16-T00, `T30/T00` = T30-T00) %>%
  melt() %>%
  separate(cond, c("genotype","nutrient","replicate"), sep = "_") 

relative_pS %>%
  dplyr::filter(variable %in% c("T00/T00","T06/T00","T16/T00","T30/T00")) %>%
  dplyr::filter(replicate == "R1") %>%
  ggplot(aes(x=variable, y=value, group=pSite, color=pSite)) +
  facet_grid(nutrient~genotype)+
  geom_line(size=0.3,alpha=0.5,colours = pSite)+
  theme_bw(base_size=10) +
  #scale_fill_brewer() +
  #scale_color_brewer() +
  ylab(expression(log[2]*" (normalized expression) ")) +
  ggtitle("Dynamics of each pSite") +  
  theme(panel.grid.major = element_blank(), panel.grid.minor =element_blank(),legend.position="none")+
  xlab("Hours post inoculation")

###################################################
#t-test between samples 
pS_reshape <- relative_pS %>% 
  dplyr::rename(time = variable) %>%
  dplyr::filter(time %in% c("T00","T06","T16","T30")) %>%
  group_by(`Proteins`, `Leading proteins`, `Protein names`,`Gene names`, pSite, `Fasta headers`, nutrient, time) %>%
  unite(cond, genotype, replicate) %>%
  spread(key = cond, value = value) 

delete.na <- function(DF, n=0) {DF[rowSums(is.na(DF)) <= n,]}

pS_filtered <- delete.na(pS_reshape,2)
  
#run t-test and log2FC
ttest_pS <- pS_filtered %>%
  melt() %>%
  separate(variable, c("genotype","replicate")) %>%
  group_by(`Proteins`, `Leading proteins`, `Protein names`,`Gene names`, pSite, `Fasta headers`, nutrient, time) %>%
  summarise_each(funs(t.test(.[genotype == "WT"], .[genotype == "rim15KO"], var.equal=TRUE)$p.value), pvalue = value) 

ttest_pS$qvalue <- p.adjust(ttest_pS$pvalue,method="BH") 

log2FC_pS <- pS_filtered %>%
  melt() %>%
  separate(variable, c("genotype","replicate")) %>%
  group_by(`Proteins`, `Leading proteins`, `Protein names`,`Gene names`, pSite, `Fasta headers`, nutrient, time) %>%
  summarise_each(funs(mean(.[genotype=="WT"], na.rm=TRUE)-mean(.[genotype == "rim15KO"], na.rm=TRUE)), log2FC = value)

logFC_ttest_pS <-left_join(ttest_pS, log2FC_pS, by = c( "Proteins", "Leading proteins", "Protein names", "Gene names","pSite","Fasta headers", "nutrient","time")) %>%
    mutate(qvalue = round(qvalue, digits = 4))

write_csv(logFC_ttest_pS,"./tables/DE_pSites_byGenotype.csv")

T00 <- logFC_ttest_pS %>% dplyr::filter(nutrient == "P", qvalue <= 0.051, abs(log2FC) > 0, time == "T00")

sample_protein <- logFC_ttest_pS %>%
  dplyr::filter(pSite == "TIF4631_163") %>%
  dplyr::filter(nutrient == "P") 

show_syn <- logFC_ttest_pS %>%
  dplyr::filter(pSite %in% c("IGO1_64","SUI3_92","TIF5_189","TIF5_190","FUN12_386","DPS1_14","TIF4631_163","DPS1_14","HCR1_65","SEC16_1760","EAP1_391")) %>%
  dplyr::filter(nutrient == "P") #%>%

show_deg<-logFC_ttest_pS %>%
  dplyr::filter(pSite %in% c("CUE5_50","YRB1_60","BRE5_282")) %>%
  dplyr::filter(nutrient == "P") #%>%
                             
show_trehelose<-logFC_ttest_pS %>% 
    dplyr::filter(pSite %in% c("TPS3_148","TSL1_73","TSL1_155","TSL1_147")) %>%
    dplyr::filter(nutrient == "P") #%>%
show <- logFC_ttest_pS %>% 
    dplyr::filter(pSite %in% c("IGO1_64","SUI3_92","TIF5_189","TIF5_190","FUN12_386","DPS1_14","TIF4631_163","DPS1_14","HCR1_65","SEC16_1760","EAP1_391", "TPS3_148","TSL1_73","TSL1_155","TSL1_147","CUE5_50","YRB1_60","BRE5_282","BLM10_62")) %>%
    dplyr::filter(nutrient == "P")
#  mutate(qvalue = round(qvalue, digits = 4))

ggplot(data=logFC_ttest_pS%>% dplyr::filter(nutrient == "P"), aes(x=log2FC, y =-log10(qvalue))) +
  geom_point(data=dplyr::filter(logFC_ttest_pS, qvalue > 0.05,nutrient == "P"), alpha = 1/6, size = 0.8) +
  geom_point(data=dplyr::filter(logFC_ttest_pS, log2FC < 0, nutrient == "P",qvalue < 0.05), alpha = 1, size = 1.5,  color = c("#00afbb")) +
  geom_point(data=dplyr::filter(logFC_ttest_pS, log2FC > 0,nutrient == "P", qvalue < 0.05), alpha = 1, size = 1.5, color = 
c("#e7b800")) +
  theme_classic() +
  coord_fixed(ratio = 6)+
  #geom_text_repel(data = dplyr::filter(logFC_ttest_pS, nutrient == "P", log2FC > 2, qvalue < 0.05), aes(label = pSite), size =2.5,hjust = 0, nudge_x = 0.05,direction = "y",segment.size = 0.2) +
 #geom_text_repel(data = dplyr::filter(logFC_ttest_pS,nutrient == "P",  log2FC < -2, qvalue < 0.05), aes(label = pSite), size =2.5,hjust = 1, nudge_x = 0.5,direction = "y",segment.size = 0.2) +
xlab(expression(Log[2]*" (WT / rim15Δ)")) +
  ylab(expression(-Log[10]* "(qvalue)")) +
  xlim(-2.5,6.5)+
  ylim(0,2)+
  theme(plot.title = element_text(hjust = 0.5, size = 14), axis.title = element_text(size = 12), axis.text = element_text(size = 10, colour = "black"))+
  facet_grid(nutrient~time)+
  theme(legend.position="none")
ggsave("./plots/pSite_volcano.pdf",width = 7, height = 4, useDingbats=FALSE)

ggplot(data=logFC_ttest_pS %>% dplyr::filter(nutrient == "P", time == "T16"), aes(x=log2FC, y =-log10(qvalue))) +
  geom_point(data=dplyr::filter(logFC_ttest_pS, qvalue > 0.05,nutrient == "P", time == "T16"), alpha = 1/6, size = 2) +
  geom_point(data=dplyr::filter(logFC_ttest_pS, log2FC < 0, qvalue <= 0.051,nutrient == "P", time == "T16"), alpha =1, size =2.5,  color = c("#00afbb")) +
  geom_point(data=dplyr::filter(logFC_ttest_pS, log2FC > 0, qvalue <= 0.051,nutrient == "P", time == "T16"), alpha = 1, size =2.5, color = 
c("#e7b800")) +
  theme_classic() +
  geom_text_repel(data = dplyr::filter(show, log2FC < 0, qvalue <= 0.051, nutrient == "P", time == "T16"), aes(label = pSite), size =3, hjust = 1, nudge_x = 0.05,direction = "y",segment.size = 0.5) +
 geom_text_repel(data = dplyr::filter(show, log2FC > 0, qvalue <= 0.051, nutrient == "P", time == "T16"), aes(label = pSite), size =3, hjust = 0, nudge_x = 0.5,direction = "y",segment.size = 0.5) +
xlab(expression(Log[2]*" (WT / rim15Δ)")) +
  ylab(expression(-Log[10]* "(qvalue)")) +
  theme(plot.title = element_text(hjust = 0.5, size = 14), axis.title = element_text(size = 14), axis.text = element_text(size = 14, colour = "black"))+
  ylim(0,2) +
  #xlim(-5,6.5)+
  facet_grid(nutrient~time)+
  theme(legend.position="none")
ggsave("./plots/pSite_volcano_P_T16.pdf", width = 7, height = 4, useDingbats=FALSE)

#look at the differentially expressed proteins 
c_pSites <- logFC_ttest_pS %>% filter(nutrient == "C",  qvalue < 0.05)
p_pSites <- logFC_ttest_pS %>% filter(nutrient == "P",  qvalue < 0.05)

c_list <- unique(c_pSites$pSite)
p_list <- unique(p_pSites$pSite)

length(intersect(c_list, p_list))

length(c_list)

length(p_list)

pSite_sum_genotype <- logFC_ttest_pS %>% 
   mutate(FC = case_when(log2FC > 0 & qvalue < 0.05 ~ "UP in WT", 
                         log2FC < 0 & qvalue < 0.05 ~ "UP in rim15KO",
                         qvalue > 0.05 ~ "neutral")) %>%
  group_by(nutrient,time,FC) %>%
  dplyr::summarise(count = n()) %>%
  filter(FC != "neutral")

pSite_sum_genotype %>%
  #filter(nutrient == "P") %>%
  transform(nutrient=factor(nutrient,levels=c("C","P"))) %>% 
  transform(FC=factor(FC,levels=c("UP in WT","UP in rim15KO"))) %>% 
  ggplot(aes(x=time, y=count, fill=FC)) +
  facet_wrap(~nutrient)+
  scale_fill_manual(values=c('#e7b800','#00afbb')) +
  geom_bar(stat="identity") +
  xlab("Hours post nutrient limitation") +
  ylab("Number of phosphorylation sites (p.adj < 0.05)")
```

#collect sequence window for significantly changed pSites estimated by ttest
```{r motif search, echo = FALSE}
ratio_sequence <- phosphosites %>%
  dplyr::select(Proteins, id, `Positions within proteins`,`Leading proteins`,`Protein names`,`Gene names`,`Fasta headers`, `Localization prob`, `Sequence window`) %>%
  dplyr::select(Proteins, id, `Positions within proteins`,`Leading proteins`,`Protein names`,`Gene names`,`Fasta headers`, `Localization prob`,`Sequence window`) %>%
  dplyr::select(Proteins, id, `Positions within proteins`,`Leading proteins`,`Protein names`,`Gene names`,`Fasta headers`, `Localization prob`, `Sequence window`) %>%
  dplyr::select(-matches("_N_")) %>%
  dplyr::select(-matches("___")) %>%
  mutate(id = factor(id)) %>%
  unite(pSite, `Gene names`, `Positions within proteins`, remove = FALSE) %>%
  dplyr::filter(`Localization prob` > 0.7) %>%
  select(-c(`Localization prob`, `Positions within proteins`)) %>%
  dplyr::select(-contains("variability")) %>%
  dplyr::select(-contains("count")) %>%
  dplyr::select(-contains("iso-count")) %>%
  dplyr::select(-contains("type")) 

n=9
m=22
##gets the phophosite and 14aa window for the background
ratio_sequence$motif <- substr(ratio_sequence$`Sequence window`, n, m)
ratio_sequence$shortmotif <- substr(ratio_sequence$`Sequence window`, 16, 17)
ratio_sequence_motif <- dplyr::filter(ratio_sequence, substring(ratio_sequence$motif,m-n+1,m-n+1) != "_")

sig_logFC_ttest_pS <- right_join(ratio_sequence,logFC_ttest_pS %>% dplyr:: filter(log2FC > 0 & qvalue < 0.05), by = c("pSite","Proteins","Leading proteins","Protein names","Gene names","Fasta headers"))

sig_logFC_ttest_pS$motif <- substr(sig_logFC_ttest_pS$`Sequence window`, n, m)
sig_logFC_ttest_pS$shortmotif <- substr(sig_logFC_ttest_pS$`Sequence window`, 16, 17)
sig_logFC_ttest_pS_motif <- dplyr::filter(sig_logFC_ttest_pS, substring(sig_logFC_ttest_pS$motif,m-n+1,m-n+1) != "_")

#maps % of SQ TQ sites in overall data
STQreturn <- function(x){
  if(x == "SQ" | x == "TQ") {
  return("S/TQ")
} else {
  return("Other")
}
}

ratio_sequence_motif$SQTQornot <- sapply(X = ratio_sequence_motif$shortmotif, FUN = "STQreturn")

slices <- table(ratio_sequence_motif$SQTQornot)
slices_labels <- paste(names(slices), "\n", round((slices/nrow(ratio_sequence_motif))*100), "%", sep="")
pie(slices, labels = slices_labels, 
   main="Distribution of Phosphosites\n with S/TQ motif (all phosphosites)", col =c("#DCDCDC", "#696969"))

##runs motifx
fg_P <- sig_logFC_ttest_pS_motif %>%
  dplyr::filter(nutrient == "P")
#for only in Phosphorous condition
#motif_P<-motifx(sig_logFC_ttest_pS_motif$motif, ratio_sequence_motif$motif, central.res = "S", min.seqs = 10, pval.cutoff = 0.05, verbose = F, perl.impl = F)
motif_P

fg_C <- sig_logFC_ttest_pS_motif %>%
  dplyr::filter(nutrient == "C")
#for only in carbon condition
motif_C<-motifx(fg_C$motif, ratio_sequence_motif$motif, central.res = "S", min.seqs = 10, pval.cutoff = 0.05, verbose = F, perl.impl = F)
motif_C

#for both
motifs<-motifx(sig_logFC_ttest_pS_motif$motif, ratio_sequence_motif$motif, central.res = "S", min.seqs = 10, pval.cutoff = 0.05, verbose = F, perl.impl = F)
motifs

#loading packages for motif visualization
#source("https://bioconductor.org/biocLite.R")
#biocLite(c("Biostrings","BiocStyle","Biobase","seqLogo","ggseqlogo"))
#library(devtools)
#install_github("kkdey/Logolas",build_vignettes = TRUE)
#library(Logolas)

#data("N_Glycosyl_sequences")
#bg <- apply(N_Glycosyl_sequences, 1, function(x) return(median(x)))
#bg <- bg/sum(bg)
#logomaker(N_Glycosyl_sequences, type = "EDLogo", bg=bg)
```

#save sequence windows and make motifs using iceLogo
```{r, echo = FALSE}
sig_logFC_ttest_pS <- right_join(ratio_sequence,logFC_ttest_pS %>% dplyr:: filter(log2FC > 1 & qvalue < 0.05), by = c("pSite","Proteins","Leading proteins","Protein names","Gene names","Fasta headers")) %>%drop_na()

write_csv(sig_logFC_ttest_pS,"./tables/motif_sig(p<0.01).csv")
```

#summarize for the significantly differential expressed pSites
```{r, echo = FALSE}
sig_pSites <- logFC_ttest_pS %>%
  filter(qvalue < 0.05) %>%
  group_by(nutrient, time) %>%
  summarize(n())
sig_pSites
```

#look at the protein dynamics in who has a significant differential expression between rim15 and wt
```{r, echo = FALSE}
proteinG<-read_csv("./tables/normed_ratio_perseus.csv") %>%
  dplyr::select(-contains("/")) #%>%
  #remove entries with any NA reads
  #drop_na()

pSite<-read_csv("./tables/normed_pSite_ratio_perseus.csv") %>%
  dplyr::select(-c("id")) %>%
  dplyr::select(-contains("/"))# %>%
  #remove entries with any NA reads
  #drop_na()

#+++++++++++++++++++++++++
# Function to calculate the mean and the standard deviation
  # for each group
#+++++++++++++++++++++++++
# data : a data frame
# varname : the name of a column containing the variable
  #to be summariezed
# groupnames : vector of column names to be used as
  # grouping variables
data_summary <- function(data, varname, groupnames){
  require(plyr)
  summary_func <- function(x, col){
    c(mean = mean(x[[col]], na.rm=TRUE),
      sd = sd(x[[col]], na.rm=TRUE))
  }
  data_sum<-ddply(data, groupnames, .fun=summary_func,
                  varname)
  data_sum <- rename(data_sum, c("mean" = varname))
 return(data_sum)
}

pSite_tidy <- melt(pSite) %>%
  separate(variable, c("genotype", "nutrient", "samplingTime", "replicate"), sep = "_")
sig_pSite <- logFC_ttest_pS %>% dplyr::filter(abs(log2FC) > 1, qvalue < 0.05)

sig_ttest_pS_P <- right_join(ratio_sequence,logFC_ttest_pS %>% dplyr::filter(qvalue < 0.05), by = c("pSite","Proteins","Leading proteins","Protein names","Gene names","Fasta headers")) %>%drop_na() %>%
  filter(nutrient == "P")
```

#GO term enrichment analysis
reference: https://learn.gencore.bio.nyu.edu/rna-seq-analysis/over-representation-analysis/
```{r fig.width=5, fig.height=2, echo = FALSE}
library(clusterProfiler)
require(DOSE)
sig_ttest_pS_P <- right_join(ratio_sequence,logFC_ttest_pS %>% dplyr::filter(qvalue < 0.05), by = c("pSite","Proteins","Leading proteins","Protein names","Gene names","Fasta headers")) %>%drop_na() %>%
  filter(nutrient == "P")

# we want the log2 fold change 
original_gene_list <- sig_ttest_pS_P$log2FC

# name the vector
names(original_gene_list) <- sig_ttest_pS_P$`Gene names`

# omit any NA values 
gene_list<-na.omit(original_gene_list)

# sort the list in decreasing order (required for clusterProfiler)
gene_list = sort(gene_list, decreasing = TRUE)

gse <- gseGO(geneList=gene_list, 
             ont ="ALL", 
             keyType = "GENENAME", 
             nPerm = 1000, 
             minGSSize = 3, 
             maxGSSize = 50, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = "org.Sc.sgd.db", 
             pAdjustMethod = "none")

dotplot(gse, showCategory=30, split=".sign")  + facet_grid(.~.sign)

#KEgg
#ids<-bitr(names(original_gene_list), fromType = "GENENAME", toType = "ENTREZID", OrgDb="org.Sc.sgd.db")

# remove duplicate IDS (here I use "ENSEMBL", but it should be whatever was selected as keyType)
#dedup_ids = ids[!duplicated(ids[c("GENENAME")]),]

# Create a new dataframe df2 which has only the genes which were successfully mapped using the bitr function above
#df2 = right_join(sig_ttest_pS_P %>% dplyr::rename(GENENAME = `Gene names`), dedup_ids, by = "GENENAME")

# Create a vector of the gene unuiverse
#kegg_gene_list <- df2$log2FC

# Name vector with ENTREZ ids
#names(kegg_gene_list) <- df2$ENTREZID

# omit any NA values 
#kegg_gene_list<-na.omit(kegg_gene_list)

# sort the list in decreasing order (required for clusterProfiler)
#kegg_gene_list = sort(kegg_gene_list, decreasing = TRUE)

#kegg_organism = "sce"
#kk2 <- gseKEGG(geneList     = kegg_gene_list,
#               organism     = kegg_organism,
#               nPerm        = 10000,
#               minGSSize    = 1,
#               maxGSSize    = 50,
               #pvalueCutoff = 0.05,
#               pAdjustMethod = "none",
#               keyType       = "ncbi-geneid")
##enrichment analysis 
#over representation test
#go_enrich <- enrichGO(gene = gene_list,
#         universe = bg,
#         OrgDb = "org.Sc.sgd.db", 
#         keyType = 'GENENAME',
#         readable = T,
#         ont = "ALL")
  
#BiocManager::install("enrichplot")
#library(enrichplot)
#upsetplot(enrich)
#wcdf<-read.table(text=go_enrich$GeneRatio, sep = "/")[1]
#wcdf$term<-go_enrich[,2]
#wordcloud(words = wcdf$term, freq = wcdf$V1, scale=(c(4, .1)), colors=brewer.pal(8, "Dark2"), max.words = 25)
```


```{r, echo = FALSE}
###############################################################################################
pSite<-read_csv("./tables/normed_pSite_ratio_perseus.csv") %>%
    dplyr::select(-c("id"))

pSite_tidy <- melt(pSite) %>%
  separate(variable, c("genotype", "nutrient", "samplingTime", "replicate"), sep = "_") %>%
  dplyr::filter(genotype %in% c("rim15KO","WT"))

#drop na function
delete.na <- function(DF, n=0) {DF[rowSums(is.na(DF)) <= n,]}

#t-test between samples 
pSite_reshape <- pSite_tidy %>% 
  dplyr::rename(time = samplingTime) %>%
  dplyr::filter(time %in% c("T00","T06","T16","T30")) %>%
  group_by(`Proteins`, pSite, `Protein names`,`Gene names`,`Fasta headers`,genotype, time) %>%
  unite(cond, nutrient, replicate) %>%
  drop_na() %>%
  spread(key = cond, value = value) 

###################################################
#remove valus whose NA value exist more than two times
pSite_filtered <- delete.na(pSite_reshape,2)

#run t-test and log2FC
ttest_pSite <- pSite_filtered %>%
  melt() %>%
  separate(variable, c("nutrient","replicate")) %>%
  group_by(pSite, `Protein names`,`Gene names`,`Fasta headers`,genotype, time) %>%
  summarise_each(funs(t.test(.[nutrient == "C"], .[nutrient == "P"], var.equal=TRUE)$p.value), pvalue = value) 

ttest_pSite$qvalue <- p.adjust(ttest_pSite$pvalue,method="BH") 

log2FC_pSite <- pSite_filtered %>%
  melt() %>%
  separate(variable, c("nutrient","replicate")) %>%
  group_by(pSite, `Protein names`,`Gene names`,`Fasta headers`, genotype, time) %>%
  summarise_each(funs(mean(.[nutrient=="C"], na.rm=TRUE)-mean(.[nutrient == "P"], na.rm=TRUE)), log2FC = value)

logFC_ttest_pSite <-left_join(ttest_pSite, log2FC_pSite, by = c("pSite","Protein names","Gene names","Fasta headers","time","genotype")) %>%
  dplyr::filter(genotype %in% c("WT","rim15KO"))

ggplot(data=logFC_ttest_pSite, aes(x=log2FC, y =-log10(qvalue))) +
  geom_point(data=dplyr::filter(logFC_ttest_pSite,  qvalue > 0.05), alpha = 1/6, size = 0.8) +
  geom_point(data=dplyr::filter(logFC_ttest_pSite, log2FC < -1, qvalue < 0.05), alpha = 1, size = 1.5,  color = c("#00afbb")) +
  geom_point(data=dplyr::filter(logFC_ttest_pSite, log2FC > 1, qvalue < 0.05), alpha = 1, size = 1.5, color = 
c("#e7b800")) +
  theme_classic() +
  #coord_fixed(ratio = 6) +
  xlab(expression(Log[2]*" (C / P)")) +
  ylab(expression(-Log[10]* "(qvalue)")) +
  xlim(-10,10) +
  #ylim(0,2) +
  theme(plot.title = element_text(hjust = 0.5, size = 14), axis.title = element_text(size = 12), axis.text = element_text(size = 10, colour = "black")) +
  facet_grid(genotype~time)+
  theme(legend.position="none")
ggsave("./plots/pSites_volcano_nutrient.pdf",width = 7, height = 4,  useDingbats=FALSE)


pSite_sum_nutrient <- logFC_ttest_pSite %>% 
   mutate(FC = case_when(log2FC > 0 & qvalue < 0.05 ~ "UP in C", 
                         log2FC < 0 & qvalue < 0.05 ~ "UP in P",
                         qvalue > 0.05 ~ "neutral")) %>%
  group_by(genotype,time,FC) %>%
  dplyr::summarise(count = n()) %>%
  filter(FC != "neutral")

pSite_sum_nutrient %>%
  transform(genotype=factor(genotype,levels=c("WT","rim15KO"))) %>% 
  ggplot(aes(x=time, y=count, fill=FC)) +
  facet_wrap(~genotype)+
  scale_fill_manual(values=c('orange','purple')) +
  geom_bar(stat="identity") +
  xlab("Hours post nutrient limitation") +
  ylab("Number of phosphorylation sites (p.adj<0.05)")
```