---
title: "linearnility check for protein and pSites"
output: html_notebook
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(collapse=TRUE, comment="##", fig.retina=2, fig.path = "README_figs/README-")
```

Load packages
```{r, include=FALSE}
library(dplyr)
library(ggplot2)
library(reshape2)
library(tidyverse)
library(GGally)
library(ggpubr)
library(ggrepel)
library(gridExtra)
library(devtools)
library(ggfortify)
library(limma)
library(plotly)
library(broom)
library(rmotifx)
library(pheatmap)
```

Read in tables
```{r}
#read in the master table: 
master <- read_csv("./tables/master_protein.csv")
```

Make plot on one example gene 
```{r}
IGO1 <- master %>% filter(`Gene names` == "IGO1") %>%
  transform(genotype=factor(genotype,levels=c("WT","rim15KO")))

#plot both on the same plot
#caculate the scalefactor
scaleFactor <- max(IGO1$normalized_ratio) / max(IGO1$normalized_intensity)
protein = "IGO1"
ggplot(IGO1, aes(x=samplingTime)) +
  geom_smooth(aes(y=normalized_ratio, group = genotype), method="lm", col = "#eb4034") +
  geom_point(aes(y=normalized_ratio), col = "#eb4034")+
  geom_smooth(aes(y=normalized_intensity * scaleFactor, group = genotype), method="lm", col = "#34a2eb") +
  geom_point(aes(y=normalized_intensity * scaleFactor), col = "#34a2eb")+
  facet_grid(nutrient~genotype, scale = "free")+
  scale_y_continuous(name="Normalized ratio", sec.axis=sec_axis(~./scaleFactor, name="Normalized intensity"))  +
  theme(axis.title.y = element_text(color = "#eb4034", size=13),
    axis.ticks.y = element_line(color = "#eb4034"),
    axis.text.y = element_text(color = "#eb4034"), 
    axis.title.y.right = element_text(color = "#34a2eb", size=13),
    axis.ticks.y.right = element_line(color = "#34a2eb"),
    axis.text.y.right = element_text(color = "#34a2eb"),
    panel.background = element_blank(),
    axis.line.x = element_line(colour = "black"),
    axis.line.y = element_line(colour = "#eb4034"),
    axis.line.y.right = element_line(colour = "#34a2eb"))+
  ggtitle(protein)
```

Map the function to each gene
```{r, include = FALSE,echo=FALSE}
proteome <- master %>% melt() %>%
  filter(variable %in% c("normalized_ratio", "normalized_intensity")) %>%
  group_by(`Gene names`,`Protein IDs`, genotype, nutrient, samplingTime, variable) %>%
  #count the number of na values
  mutate(ind = sum(is.na(value))) %>% 
  #remove the proteins who has NA value in all conditions and samples
  dplyr::filter(!any(ind > 1)) %>%
    #remove infinite values
  #filter(!is.infinite(value)) %>%
  select(-ind) %>% 
  spread(key = variable, value = value)

#make a delet.na function to remove na values for each row with a certain number  
delete.na <- function(DF, n=0) {DF[rowSums(is.na(DF)) <= n,]}
proteome <- delete.na(proteome)

protein_list <- unique(proteome$`Protein IDs`)
plot_list = list()
for (i in 1:length(protein_list)) {
  protein <- proteome %>% filter(`Protein IDs` == protein_list[i]) %>%
  transform(genotype=factor(genotype,levels=c("WT","rim15KO")))
  
  scaleFactor <- max(protein$normalized_ratio, na.rm = TRUE) / max(protein$normalized_intensity)
  
  p <- ggplot(protein, aes(x=samplingTime)) +
  geom_smooth(aes(y=normalized_ratio, group = genotype), method="lm", col = "#eb4034") +
  geom_point(aes(y=normalized_ratio), col = "#eb4034")+
  geom_smooth(aes(y=normalized_intensity * scaleFactor, group = genotype), method="lm", col = "#34a2eb") +
  geom_point(aes(y=normalized_intensity * scaleFactor), col = "#34a2eb")+
  facet_grid(nutrient~genotype, scale = "free")+
  scale_y_continuous(name="Normalized ratio", sec.axis=sec_axis(~./scaleFactor, name="Normalized intensity"))  +
  theme(axis.title.y = element_text(color = "#eb4034", size=13),
    axis.ticks.y = element_line(color = "#eb4034"),
    axis.text.y = element_text(color = "#eb4034"), 
    axis.title.y.right = element_text(color = "#34a2eb", size=13),
    axis.ticks.y.right = element_line(color = "#34a2eb"),
    axis.text.y.right = element_text(color = "#34a2eb"),
    panel.background = element_blank(),
    axis.line.x = element_line(colour = "black"),
    axis.line.y = element_line(colour = "#eb4034"),
    axis.line.y.right = element_line(colour = "#34a2eb"))+
  ggtitle(unique(protein$Gene.names))

  plot_list[[i]] = p
}

#check for the 
#plot_list[sapply(plot_list, Negate(anyNA))]

ggsave(filename = "./plots/individual_plots_protein_new.pdf", plot = marrangeGrob(plot_list, nrow=4, ncol=3), width = 15, height = 10)
```
